From 553c8fd22b2562044db7e0aa551138b4c2bbb290 Mon Sep 17 00:00:00 2001
From: Ignacio Sanchez Gines <863613+drhelius@users.noreply.github.com>
Date: Sat, 10 May 2025 01:37:54 +0200
Subject: [PATCH] [libretro] Update info file

---
 platforms/libretro/geargrafx_libretro.info | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/platforms/libretro/geargrafx_libretro.info b/platforms/libretro/geargrafx_libretro.info
index 1ac8eee..b2bb4bc 100644
--- a/platforms/libretro/geargrafx_libretro.info
+++ b/platforms/libretro/geargrafx_libretro.info
@@ -3,10 +3,10 @@ display_name = "NEC - PC Engine / SuperGrafx (Geargrafx)"
 authors = "Ignacio Sanchez"
 supported_extensions = "pce|sgx|bin|rom"
 corename = "Geargrafx"
+categories = "Emulator"
 license = "GPLv3"
 permissions = ""
-display_version = "1.0.0"
-categories = "Emulator"
+display_version = "1.4.0"
 
 # Hardware Information
 manufacturer = "NEC"
@@ -14,7 +14,9 @@ systemname = "PC Engine/SuperGrafx"
 systemid = "pc_engine"
 
 # Libretro Features
-database = "NEC - PC Engine SuperGrafx|NEC - PC Engine - TurboGrafx 16"
 supports_no_game = "false"
+database = "NEC - PC Engine SuperGrafx|NEC - PC Engine - TurboGrafx 16"
 savestate = "true"
 savestate_features = "deterministic"
+
+description = "Libretro version of the accurate Geargrafx emulator. This core supports PC Engine, TurboGrafx and SuperGrafx consoles. It faithfully emulates all of the hardware and is an excellent choice for emulating these consoles on a modern system."
\ No newline at end of file
