---
 src/joystick/SDL_joystick.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 0d7fa7a..4fafcd2 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1494,7 +1494,7 @@ int SDL_JoystickRumble(SDL_Joystick *joystick, Uint16 low_frequency_rumble, Uint
             } else {
                 joystick->rumble_resend = 0;
             }
-        }
+
 
         if (retval == 0) {
             joystick->low_frequency_rumble = low_frequency_rumble;
@@ -1510,6 +1510,8 @@ int SDL_JoystickRumble(SDL_Joystick *joystick, Uint16 low_frequency_rumble, Uint
                 joystick->rumble_resend = 0;
             }
         }
+
+	}
     }
     SDL_UnlockJoysticks();
 
-- 
2.47.1
