---
 src/joystick/SDL_joystick.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index b472777..37f7b32 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1688,6 +1688,7 @@ void SDL_JoystickQuit(void)
 
 static SDL_bool SDL_PrivateJoystickShouldIgnoreEvent(void)
 {
+    return SDL_FALSE;
     if (SDL_joystick_allows_background_events) {
         return SDL_FALSE;
     }
-- 
2.47.1
