diff --git a/package/libcec/libcec.mk b/package/libcec/libcec.mk
index 33e0463eb8..33b70eeb7d 100644
--- a/package/libcec/libcec.mk
+++ b/package/libcec/libcec.mk
@@ -37,6 +37,16 @@ ifeq ($(BR2_PACKAGE_XLIB_LIBXRANDR),y)
 LIBCEC_DEPENDENCIES += xlib_libXrandr
 endif
 
+# reglinux
+ifeq ($(BR2_PACKAGE_LIBCEC_LINUX_API),y)
+LIBCEC_CONF_OPTS += -DHAVE_LINUX_API=1
+endif
+
+# reglinux
+ifeq ($(BR2_PACKAGE_SYSTEM_TARGET_AMLOGIC_ANY),y)
+LIBCEC_CONF_OPTS += -DHAVE_AMLOGIC_API=1
+endif
+
 # Disable information about how libCEC is compiled.
 LIBCEC_CONF_OPTS += -DHAVE_GIT_BIN="" \
 	-DHAVE_WHOAMI_BIN="" \
