diff --git a/package/qt6/Config.in b/package/qt6/Config.in
index c5a1a702d3..4cc6cf325d 100644
--- a/package/qt6/Config.in
+++ b/package/qt6/Config.in
@@ -10,6 +10,8 @@ config BR2_PACKAGE_QT6_ARCH_SUPPORTS
 	# sparc 32-bit is supposedly supported, but has issues due to
 	# the need of libatomic
 	default y if BR2_sparc64
+	# REGLINUX ADD RISC-V 64-bit support
+	default y if BR2_riscv && BR2_ARCH_IS_64
 	depends on BR2_USE_MMU
 
 config BR2_PACKAGE_QT6_GL_SUPPORTS
