diff --git a/package/qt6/Config.in b/package/qt6/Config.in
index 10594691cf..34b1e269bd 100644
--- a/package/qt6/Config.in
+++ b/package/qt6/Config.in
@@ -7,6 +7,7 @@ config BR2_PACKAGE_QT6_ARCH_SUPPORTS
 	default y if BR2_x86_64
 	default y if BR2_mips || BR2_mipsel || BR2_mips64 || BR2_mips64el
 	default y if BR2_powerpc || BR2_powerpc64 || BR2_powerpc64le
+	default y if BR2_riscv && BR2_ARCH_IS_64 # reglinux add RISC-V 64-bit support
 	# sparc 32-bit is supposedly supported, but has issues due to
 	# the need of libatomic
 	default y if BR2_sparc64
