diff --git a/package/alsa-lib/Config.in b/package/alsa-lib/Config.in
index 360122f97f..304f947f83 100644
--- a/package/alsa-lib/Config.in
+++ b/package/alsa-lib/Config.in
@@ -73,11 +73,6 @@ config BR2_PACKAGE_ALSA_LIB_UCM
 	depends on BR2_USE_MMU # fork()
 	select BR2_PACKAGE_ALSA_LIB_MIXER
 
-config BR2_PACKAGE_ALSA_LIB_ALISP
-	bool "alisp"
-	default y
-	depends on BR2_USE_MMU
-
 config BR2_PACKAGE_ALSA_LIB_OLD_SYMBOLS
 	bool "old-symbols"
 	default y
