--- a/src/lj_arch.h	2024-03-12 10:25:24.000000000 +0100
+++ b/src/lj_arch.h	2024-06-19 23:12:43.926835913 +0200
@@ -444,7 +444,6 @@
 #endif
 
 #elif LUAJIT_TARGET == LUAJIT_ARCH_RISCV64
-#if defined(__riscv_float_abi_double)
 
 #define LJ_ARCH_NAME		"riscv64"
 #define LJ_ARCH_BITS		64
@@ -461,10 +460,6 @@
 #define LJ_ARCH_NUMMODE		LJ_NUMMODE_DUAL
 
 #else
-#error "No support for RISC-V 64 Soft-float/Single-float"
-#endif
-
-#else
 #error "No target architecture defined"
 #endif
 
@@ -557,9 +552,6 @@
 #undef LJ_TARGET_MIPS
 #endif
 #elif LJ_TARGET_RISCV64
-#if !defined(__riscv_float_abi_double)
-#error "Only RISC-V 64 double float supported for now"
-#endif
 #endif
 #endif
 
