diff --git a/package/spirv-llvm-translator/spirv-llvm-translator.mk b/package/spirv-llvm-translator/spirv-llvm-translator.mk
index 523890a986e..8daf7e9b690 100644
--- a/package/spirv-llvm-translator/spirv-llvm-translator.mk
+++ b/package/spirv-llvm-translator/spirv-llvm-translator.mk
@@ -5,12 +5,27 @@
 ################################################################################
 
 # Generate version string using:
-#   git describe --tags --match 'v15*' --abbrev=40 origin/llvm_release_150
-SPIRV_LLVM_TRANSLATOR_VERSION = v15.0.0-46-ge82ecc2bd7295604fcf1824e47c95fa6a09c6e63
+#   git describe --tags --match 'v18*' --abbrev=40 origin/llvm_release_180
+# reglinux - 19.1.4 update
+SPIRV_LLVM_TRANSLATOR_VERSION = v19.1.4
 SPIRV_LLVM_TRANSLATOR_SITE = $(call github,KhronosGroup,SPIRV-LLVM-Translator,$(SPIRV_LLVM_TRANSLATOR_VERSION))
 SPIRV_LLVM_TRANSLATOR_LICENSE = NCSA
 SPIRV_LLVM_TRANSLATOR_LICENSE_FILES = LICENSE.TXT
-HOST_SPIRV_LLVM_TRANSLATOR_DEPENDENCIES = host-clang host-llvm host-spirv-headers
+
+# reglinux we need target package for some mesa targets
+SPIRV_LLVM_TRANSLATOR_INSTALL_STAGING = YES
+SPIRV_LLVM_TRANSLATOR_DEPENDENCIES = host-spirv-llvm-translator llvm clang spirv-headers
+SPIRV_LLVM_TRANSLATOR_CONF_OPTS = \
+	-DLLVM_BUILD_TOOLS=ON \
+	-DLLVM_DIR=$(STAGING_DIR)/lib/cmake/llvm \
+	-DLLVM_SPIRV_BUILD_EXTERNAL=YES \
+	-DLLVM_SPIRV_INCLUDE_TESTS=OFF \
+	-DLLVM_EXTERNAL_PROJECTS="SPIRV-Headers" \
+	-DLLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=$(STAGING_DIR)/include
+$(eval $(cmake-package))
+
+# reglinux define both host and targets
+HOST_SPIRV_LLVM_TRANSLATOR_DEPENDENCIES = llvm clang host-spirv-headers
 HOST_SPIRV_LLVM_TRANSLATOR_CONF_OPTS = \
 	-DLLVM_BUILD_TOOLS=ON \
 	-DLLVM_DIR=$(HOST_DIR)/lib/cmake/llvm \
@@ -19,4 +34,26 @@ HOST_SPIRV_LLVM_TRANSLATOR_CONF_OPTS = \
 	-DLLVM_EXTERNAL_PROJECTS="SPIRV-Headers" \
 	-DLLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=$(HOST_DIR)/include
 
+# reglinux fix for prebuilt llvm/clang
+#HOST_SPIRV_LLVM_TRANSLATOR_CONF_OPTS += -DCMAKE_INSTALL_RPATH="$(HOST_DIR)/lib"
+# -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+HOST_SPIRV_LLVM_TRANSLATOR_DEPENDENCIES += host-patchelf
+define HOST_SPIRV_LLVM_TRANSLATOR_FIX_RPATH
+    for f in $(HOST_DIR)/bin/llvm-* $(HOST_DIR)/bin/*ll*; do \
+        $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $$f || true; \
+    done
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/obj2yaml || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/yaml2obj || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/opt || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/sancov || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/verify-uselistorder || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/dsymutil || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/sanstats || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/reduce-chunk-list || true;
+    $(HOST_DIR)/bin/patchelf --set-rpath '$$ORIGIN/../lib' $(HOST_DIR)/bin/bugpoint || true;
+endef
+
+HOST_SPIRV_LLVM_TRANSLATOR_POST_INSTALL_HOOKS += HOST_SPIRV_LLVM_TRANSLATOR_FIX_RPATH
+
 $(eval $(host-cmake-package))
+
