--- a/core/rend/gles/gltex.cpp	2024-04-08 21:19:58.878724011 +0200
+++ b/core/rend/gles/gltex.cpp	2024-04-08 21:23:53.861586144 +0200
@@ -30,13 +30,21 @@
 	case TextureType::_8888:
 		bytesPerPixel = 4;
 		gltype = GL_UNSIGNED_BYTE;
+#ifdef LIBRETRO
+		internalFormat = GL_RGBA8_OES;
+#else
 		internalFormat = GL_RGBA8;
+#endif
 		break;
 	case TextureType::_8:
 		bytesPerPixel = 1;
 		gltype = GL_UNSIGNED_BYTE;
 		comps = gl.single_channel_format;
+#ifdef LIBRETRO
+ 		internalFormat = GL_R8_EXT;
+#else
		internalFormat = GL_R8;
+#endif
 		break;
 	default:
 		die("Unsupported texture type");
