diff --git a/support/scripts/mkusers b/support/scripts/mkusers
index 87a36a89a1..59200a365c 100755
--- a/support/scripts/mkusers
+++ b/support/scripts/mkusers
@@ -385,12 +385,11 @@ add_one_user() {
         done
     fi
 
-    # If the user has a home, chown it
-    # (Note: stdout goes to the fakeroot-script)
-    if [ "${home}" != "-" ]; then
-        mkdir -p "${TARGET_DIR}/${home}"
-        printf "chown -h -R %d:%d '%s'\n" "${uid}" "${_gid}" "${TARGET_DIR}/${home}"
-    fi
+    # reglinux : /userdata must be empty while it will be mounted over (and fuse exfat doesn't like it)
+    #if [ "${home}" != "-" ]; then
+    #    mkdir -p "${TARGET_DIR}/${home}"
+    #    printf "chown -h -R %d:%d '%s'\n" "${uid}" "${_gid}" "${TARGET_DIR}/${home}"
+    #fi
 }
 
 #----------------------------------------------------------------------------
@@ -450,7 +449,7 @@ main() {
                 auto_id=${AUTO_USER_ID}
             fi
             for g in ${groups//,/ }; do
-                add_one_group "${g}" "${auto_id}"
+                add_one_group "${g}" ${auto_id}
             done
         fi
     done
