--- a/src/frontend/mame/ui/ui.cpp	Wed May 26 15:42:03 2021
+++ b/src/frontend/mame/ui/ui.cpp	Sun Oct  3 15:45:07 2021
@@ -644,6 +644,8 @@
 	if (!first_time || (str > 0 && str < 60*5) || &machine().system() == &GAME_NAME(___empty) || (machine().debug_flags & DEBUG_FLAG_ENABLED) || video_none)
 		show_gameinfo = show_warnings = false;
 
+	show_gameinfo = show_warnings = false;
+
 #if defined(__EMSCRIPTEN__)
 	// also disable for the JavaScript port since the startup screens do not run asynchronously
 	show_gameinfo = show_warnings = false;
