diff -u -r output/bcm2711/build/melonds-d4e51f80601f57399db49f1010c45427bd2bf3c4/src/frontend/duckstation/gl/context.cpp melonds-wayland/src/frontend/duckstation/gl/context.cpp
--- a/src/frontend/duckstation/gl/context.cpp	2023-10-11 17:20:05.000000000 +0200
+++ b/src/frontend/duckstation/gl/context.cpp	2024-04-17 19:00:38.495563266 +0200
@@ -17,10 +17,11 @@
 #else
 #ifdef WAYLAND_ENABLED
 #include "context_egl_wayland.h"
-#endif
+#else
 #include "context_egl_x11.h"
 #include "context_glx.h"
 #endif
+#endif
 
 namespace GL {
 
@@ -73,14 +74,14 @@
 #elif defined(__APPLE__)
   context = ContextAGL::Create(wi, versions_to_try, num_versions_to_try);
 #else
-  if (wi.type == WindowInfo::Type::X11)
+  /*if (wi.type == WindowInfo::Type::X11)
   {
     const char* use_egl_x11 = std::getenv("USE_EGL_X11");
     if (use_egl_x11 && std::strcmp(use_egl_x11, "1") == 0)
       context = ContextEGLX11::Create(wi, versions_to_try, num_versions_to_try);
     else
       context = ContextGLX::Create(wi, versions_to_try, num_versions_to_try);
-  }
+  }*/
 
 #ifdef WAYLAND_ENABLED
   if (wi.type == WindowInfo::Type::Wayland)
diff -u -r output/bcm2711/build/melonds-d4e51f80601f57399db49f1010c45427bd2bf3c4/src/frontend/qt_sdl/CMakeLists.txt melonds-wayland/src/frontend/qt_sdl/CMakeLists.txt
--- a/src/frontend/qt_sdl/CMakeLists.txt	2023-10-11 17:20:05.000000000 +0200
+++ b/frontend/qt_sdl/CMakeLists.txt	2024-04-17 19:00:38.495563266 +0200
@@ -108,7 +108,7 @@
         ../duckstation/gl/context_agl.mm
     )
 else()
-    find_package(X11 REQUIRED)
+    #find_package(X11 REQUIRED)
     find_package(EGL REQUIRED)
 
     option(ENABLE_WAYLAND "Enable Wayland support" ON)
@@ -147,15 +147,11 @@
 
     target_sources(melonDS PRIVATE
         ../duckstation/gl/context_egl.cpp
-        ../duckstation/gl/context_egl_x11.cpp
-        ../duckstation/gl/context_glx.cpp
-        ../duckstation/gl/x11_window.cpp
-
+        ../duckstation/gl/context_egl_wayland.cpp
         ../glad/glad_egl.c
         ../glad/glad_glx.c
     )
-    target_link_libraries(melonDS PRIVATE "${X11_LIBRARIES}" "${EGL_LIBRARIES}")
-    target_include_directories(melonDS PRIVATE "${X11_INCLUDE_DIR}")
+    target_link_libraries(melonDS PRIVATE "${EGL_LIBRARIES}")
     add_compile_definitions(QAPPLICATION_CLASS=QApplication)
 endif()
 
