--- a/xenia-build	2025-02-02 16:41:21.164034956 +0100
+++ b/xenia-build	2025-02-02 16:42:34.942669846 +0100
@@ -851,14 +851,15 @@
                 f"-DCMAKE_BUILD_TYPE={args['config'].title()}",
                 f"-DCMAKE_C_COMPILER={os.environ.get('CC', 'clang')}",
                 f"-DCMAKE_CXX_COMPILER={os.environ.get('CXX', 'clang++')}",
-                "-GNinja"
+                '-DCMAKE_MAKE_PROGRAM=%s' % (os.environ["NINJA"]),
+                 '-GNinja'
             ] + pass_args, env=dict(os.environ))
             print("")
             if result != 0:
                 print("ERROR: cmake failed with one or more errors.")
                 return result
             result = subprocess.call([
-                    "ninja",
+                    os.environ["NINJA"],
                     f"-Cbuild/build_{args['config']}",
                 ] + pass_args, env=dict(os.environ))
             if result != 0:
