--- a/xenia-build.py	2025-11-27 15:20:05.640426412 +0100
+++ b/xenia-build.py	2025-11-27 15:21:06.634467241 +0100
@@ -990,14 +990,15 @@
                 f"-DCMAKE_BUILD_TYPE={cmake_build_type}",
                 f"-DCMAKE_C_COMPILER={cc_compiler}",
                 f"-DCMAKE_CXX_COMPILER={cxx_compiler}",
-                "-GNinja"
+                '-DCMAKE_MAKE_PROGRAM=%s' % (os.environ["NINJA"]),
+                '-GNinja'
             ] + pass_args, env=dict(os.environ))
             print("")
             if result != 0:
                 print("ERROR: cmake failed with one or more errors.")
                 return result
             result = subprocess.call([
-                    "ninja",
+                    os.environ["NINJA"],
                     f"-Cbuild/build_{args['config']}",
                 ] + pass_args, env=dict(os.environ))
             if result != 0:
