--- a/TheForceEngine/TFE_ForceScript/scriptInterface.cpp	2025-02-24 22:06:46.415707951 +0100
+++ b/TheForceEngine/TFE_ForceScript/scriptInterface.cpp	2025-02-24 22:07:51.696476865 +0100
@@ -67,7 +67,7 @@
 			case API_LEVEL_EDITOR:
 			{
 				registerSharedScriptAPI(api);
-				LevelEditor::registerScriptFunctions(api);
+				//LevelEditor::registerScriptFunctions(api);
 			} break;
 			case API_SYSTEM_UI:
 			{
