--- a/backends/imgui/misc/freetype/imgui_freetype.cpp	2024-12-10 14:08:55.612295391 +0100
+++ b/backends/imgui/misc/freetype/imgui_freetype.cpp	2024-12-10 14:09:23.611850560 +0100
@@ -430,6 +430,9 @@
     ImBitVector         GlyphsSet;          // This is used to resolve collision when multiple sources are merged into a same destination font.
 };
 
+#pragma GCC push_options
+#pragma GCC optimize ("O0")
+
 bool ImFontAtlasBuildWithFreeTypeEx(FT_Library ft_library, ImFontAtlas* atlas, unsigned int extra_flags)
 {
     IM_ASSERT(atlas->ConfigData.Size > 0);
@@ -751,6 +754,8 @@
     return true;
 }
 
+#pragma GCC pop_options
+
 // FreeType memory allocation callbacks
 static void* FreeType_Alloc(FT_Memory /*memory*/, long size)
 {
