--- a/Makefiles/Rules.inc	2025-01-23 13:40:03.922775522 +0100
+++ b/Makefiles/Rules.inc	2025-01-23 15:02:42.923563618 +0100
@@ -76,6 +76,11 @@
 
 
 #
+# GCC 14 workarounnds
+#
+SUPERMODEL_BUILD_FLAGS += -Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types
+
+#
 # Construct the compiler (C and C++) and linker flags
 #
 COMMON_CFLAGS = -c $(ARCH) $(OPT) $(WARN) $(addprefix -I,$(sort $(INCLUDE_DIRS))) -DGLEW_STATIC $(SUPERMODEL_BUILD_FLAGS)
