--- a/src/CMakeLists.txt	2024-04-02 18:28:45.000000000 +0200
+++ b/src/CMakeLists.txt	2024-04-03 14:42:53.605350617 +0200
@@ -117,7 +117,7 @@
 
     if (NOT CMAKE_BUILD_TYPE STREQUAL Debug)
         # _FORTIFY_SOURCE can't be used without optimizations.
-        add_compile_options(-Wp,-D_FORTIFY_SOURCE=2)
+        #  add_compile_options(-Wp,-D_FORTIFY_SOURCE=2)
     endif()
 
     if (CITRA_WARNINGS_AS_ERRORS)
