--- a/CMakeLists.txt	2024-04-03 14:52:45.404905421 +0200
+++ b/CMakeLists.txt	2024-04-03 14:52:48.480990240 +0200
@@ -89,7 +89,7 @@
 CMAKE_DEPENDENT_OPTION(COMPILE_WITH_DWARF "Add DWARF debugging information" ${IS_DEBUG_BUILD} "MINGW" OFF)
 option(ENABLE_LTO "Enable link time optimization" ${DEFAULT_ENABLE_LTO})
 option(CITRA_USE_PRECOMPILED_HEADERS "Use precompiled headers" ON)
-option(CITRA_WARNINGS_AS_ERRORS "Enable warnings as errors" ON)
+option(CITRA_WARNINGS_AS_ERRORS "Enable warnings as errors" OFF)
 
 include(CitraHandleSystemLibs)
 
