diff -rupbN linux.orig/drivers/hwmon/pwm-fan.c linux/drivers/hwmon/pwm-fan.c
--- linux.orig/drivers/hwmon/pwm-fan.c	2024-10-19 13:35:29.516991617 +0000
+++ linux/drivers/hwmon/pwm-fan.c	2024-11-18 15:41:47.343729846 +0000
@@ -560,7 +560,7 @@
 	if (ctx->pwm_fan_cooling_levels)
 		initial_pwm = ctx->pwm_fan_cooling_levels[ctx->pwm_fan_max_state];
 	else
-		initial_pwm = MAX_PWM;
+		initial_pwm = MAX_PWM / 3;
 
 	/*
 	 * Set duty cycle to maximum allowed and enable PWM output as well as
