--- a/Source/Core/Core/IOS/FS/FileSystemProxy.cpp	2025-03-08 23:50:32.622536271 +0100
+++ b/Source/Core/Core/IOS/FS/FileSystemProxy.cpp	2025-03-08 23:51:18.712064855 +0100
@@ -54,7 +54,7 @@
 }
 
 template <typename... Args>
-static void LogResult(ResultCode code, std::string_view format, Args&&... args)
+static void LogResult(ResultCode code, fmt::format_string<Args...> format, Args&&... args)
 {
   const std::string command = fmt::format(format, std::forward<Args>(args)...);
   GENERIC_LOG(Common::Log::IOS_FS,
@@ -63,7 +63,7 @@
 }
 
 template <typename T, typename... Args>
-static void LogResult(const Result<T>& result, std::string_view format, Args&&... args)
+static void LogResult(const Result<T>& result, fmt::format_string<Args...> format, Args&&... args)
 {
   const auto result_code = result.Succeeded() ? ResultCode::Success : result.Error();
   LogResult(result_code, format, std::forward<Args>(args)...);
