--- a/pcsx2/GS/GSVector4i_arm64.h
+++ b/pcsx2/GS/GSVector4i_arm64.h
@@ -373,7 +373,7 @@
 
 	__forceinline u32 minv_u32() const
 	{
-		return vminvq_u32(v4s);
+		return vminvq_u32(vreinterpretq_u32_s32(v4s));
 	}
 
 	__forceinline u32 maxv_s32() const
@@ -383,7 +383,7 @@
 
 	__forceinline u32 maxv_u32() const
 	{
-		return vmaxvq_u32(v4s);
+		return vmaxvq_u32(vreinterpretq_u32_s32(v4s));
 	}
 
 	__forceinline static int min_i16(int a, int b)
@@ -691,7 +691,7 @@
 
 	__forceinline GSVector4i srl16(s32 i) const
 	{
-		return GSVector4i(vreinterpretq_s32_u16(vshlq_u16(vreinterpretq_u16_s32(v4s), vdupq_n_u16(-i))));
+		return GSVector4i(vreinterpretq_s32_u16(vshlq_u16(vreinterpretq_u16_s32(v4s), vdupq_n_s16(-i))));
 	}
 
 	__forceinline GSVector4i srlv16(const GSVector4i& v) const
@@ -754,7 +754,7 @@
 
 	__forceinline GSVector4i srav32(const GSVector4i& v) const
 	{
-		return GSVector4i(vshlq_s32(vreinterpretq_u32_s32(v4s), vnegq_s32(v.v4s)));
+		return GSVector4i(vshlq_s32(v4s, vnegq_s32(v.v4s)));
 	}
 
 	template <int i>
@@ -765,7 +765,7 @@
 
 	__forceinline GSVector4i sll64(s32 i) const
 	{
-		return GSVector4i(vreinterpretq_s32_s64(vshlq_s64(vreinterpretq_s64_s32(v4s), vdupq_n_s16(i))));
+		return GSVector4i(vreinterpretq_s32_s64(vshlq_s64(vreinterpretq_s64_s32(v4s), vdupq_n_s64(i))));
 	}
 
 	__forceinline GSVector4i sllv64(const GSVector4i& v) const
@@ -781,7 +781,7 @@
 
 	__forceinline GSVector4i sra64(s32 i) const
 	{
-		return GSVector4i(vreinterpretq_s32_s64(vshlq_s64(vreinterpretq_s64_s32(v4s), vdupq_n_s16(-i))));
+		return GSVector4i(vreinterpretq_s32_s64(vshlq_s64(vreinterpretq_s64_s32(v4s), vdupq_n_s64(-i))));
 	}
 
 	__forceinline GSVector4i srav64(const GSVector4i& v) const
@@ -798,7 +798,7 @@
 
 	__forceinline GSVector4i srl64(s32 i) const
 	{
-		return GSVector4i(vreinterpretq_s32_u64(vshlq_u64(vreinterpretq_u64_s32(v4s), vdupq_n_u16(-i))));
+		return GSVector4i(vreinterpretq_s32_u64(vshlq_u64(vreinterpretq_u64_s32(v4s), vdupq_n_s64(-i))));
 	}
 
 	__forceinline GSVector4i srlv64(const GSVector4i& v) const
@@ -838,7 +838,7 @@
 		//return GSVector4i(vreinterpretq_s32_s16(vpaddq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
 		const int16x8_t a = vreinterpretq_s16_s32(v4s);
 		const int16x8_t b = vreinterpretq_s16_s32(v.v4s);
-		return GSVector4i(vqaddq_s16(vuzp1q_s16(a, b), vuzp2q_s16(a, b)));
+		return GSVector4i(vreinterpretq_s32_s16(vqaddq_s16(vuzp1q_s16(a, b), vuzp2q_s16(a, b))));
 	}
 
 	__forceinline GSVector4i addus8(const GSVector4i& v) const
@@ -1004,62 +1004,62 @@
 
 	__forceinline GSVector4i gt8(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s8(vcgtq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u8(vcgtq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i gt16(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s16(vcgtq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u16(vcgtq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i gt32(const GSVector4i& v) const
 	{
-		return GSVector4i(vcgtq_s32(v4s, v.v4s));
+		return GSVector4i(vreinterpretq_s32_u32(vcgtq_s32(v4s, v.v4s)));
 	}
 
 	__forceinline GSVector4i ge8(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s8(vcgeq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u8(vcgeq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i ge16(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s16(vcgeq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u16(vcgeq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i ge32(const GSVector4i& v) const
 	{
-		return GSVector4i(vcgeq_s32(v4s, v.v4s));
+		return GSVector4i(vreinterpretq_s32_u32(vcgeq_s32(v4s, v.v4s)));
 	}
 
 	__forceinline GSVector4i lt8(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s8(vcltq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u8(vcltq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i lt16(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s16(vcltq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u16(vcltq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i lt32(const GSVector4i& v) const
 	{
-		return GSVector4i(vcltq_s32(v4s, v.v4s));
+		return GSVector4i(vreinterpretq_s32_u32(vcltq_s32(v4s, v.v4s)));
 	}
 
 	__forceinline GSVector4i le8(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s8(vcleq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u8(vcleq_s8(vreinterpretq_s8_s32(v4s), vreinterpretq_s8_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i le16(const GSVector4i& v) const
 	{
-		return GSVector4i(vreinterpretq_s32_s16(vcleq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
+		return GSVector4i(vreinterpretq_s32_u16(vcleq_s16(vreinterpretq_s16_s32(v4s), vreinterpretq_s16_s32(v.v4s))));
 	}
 
 	__forceinline GSVector4i le32(const GSVector4i& v) const
 	{
-		return GSVector4i(vcleq_s32(v4s, v.v4s));
+		return GSVector4i(vreinterpretq_s32_u32(vcleq_s32(v4s, v.v4s)));
 	}
 
 
@@ -1087,7 +1087,7 @@
 	__forceinline bool allfalse() const
 	{
 		// MSB should be clear in all 8-bit lanes.
-		return (vmaxvq_u32(vreinterpretq_u8_s32(v4s)) & 0x80) != 0x80;
+		return (vmaxvq_u8(vreinterpretq_u8_s32(v4s)) & 0x80) != 0x80;
 	}
 
 	template <int i>
--- a/pcsx2/GS/GSVector4_arm64.h
+++ b/pcsx2/GS/GSVector4_arm64.h
@@ -703,17 +703,17 @@
 
 	__forceinline GSVector4 mul64(const GSVector4& v) const
 	{
-		return GSVector4(vmulq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s)));
+		return GSVector4(vreinterpretq_f32_f64(vmulq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s))));
 	}
 
 	__forceinline GSVector4 add64(const GSVector4& v) const
 	{
-		return GSVector4(vaddq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s)));
+		return GSVector4(vreinterpretq_f32_f64(vaddq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s))));
 	}
 
 	__forceinline GSVector4 sub64(const GSVector4& v) const
 	{
-		return GSVector4(vsubq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s)));
+		return GSVector4(vreinterpretq_f32_f64(vsubq_f64(vreinterpretq_f64_f32(v4s), vreinterpretq_f64_f32(v.v4s))));
 	}
 
 	__forceinline static GSVector4 f32to64(const GSVector4& v)
@@ -728,7 +728,7 @@
 
 	__forceinline GSVector4i f64toi32(bool truncate = true) const
 	{
-		const float64x2_t r = truncate ? v4s : vrndiq_f64(vreinterpretq_f64_f32(v4s));
+		const float64x2_t r = truncate ? vreinterpretq_f64_f32(v4s) : vrndiq_f64(vreinterpretq_f64_f32(v4s));
 		const s32 low = static_cast<s32>(vgetq_lane_f64(r, 0));
 		const s32 high = static_cast<s32>(vgetq_lane_f64(r, 1));
 		return GSVector4i(vsetq_lane_s32(high, vsetq_lane_s32(low, vdupq_n_s32(0), 0), 1));
@@ -782,6 +782,6 @@
 
 	__forceinline static GSVector4 broadcast64(const void* f)
 	{
-		return GSVector4(vreinterpretq_f64_f32(vld1q_dup_f64((const double*)f)));
+		return GSVector4(vreinterpretq_f32_f64(vld1q_dup_f64((const double*)f)));
 	}
 };
--- a/pcsx2/GS/GSVector.h
+++ b/pcsx2/GS/GSVector.h
@@ -116,7 +116,7 @@
 	m = truncate ? _mm_cvttps_epi32(v) : _mm_cvtps_epi32(v);
 #elif defined(_M_ARM64)
 	// GS thread uses default (nearest) rounding.
-	v4s = truncate ? vcvtq_s32_f32(v.v4s) : vcvtnq_u32_f32(v.v4s);
+	v4s = truncate ? vcvtq_s32_f32(v.v4s) : vreinterpretq_s32_u32(vcvtnq_u32_f32(v.v4s));
 #endif
 }
 
--- a/pcsx2/Gif_Unit.h
+++ b/pcsx2/Gif_Unit.h
@@ -123,7 +123,7 @@
 		u64 REGS64;
 		std::memcpy(&REGS64, tag.REGS, sizeof(u64));
 		REGS64 &= (0xFFFFFFFFFFFFFFFFULL >> (64 - nRegs * 4));
-		uint8x16_t vregs = vsetq_lane_u64(REGS64, vdupq_n_u64(0), 0);
+		uint8x16_t vregs = vreinterpretq_u8_u64(vsetq_lane_u64(REGS64, vdupq_n_u64(0), 0));
 
 		// get upper nibbles, interleave with lower nibbles, clear upper bits from low nibbles
 		vregs = vandq_u8(vzip1q_u8(vregs, vshrq_n_u8(vregs, 4)), vdupq_n_u8(0x0F));
