diff --git a/package/urandom-scripts/S01seedrng b/package/urandom-scripts/S01seedrng
index 0fea0bea83..f8c9169b4f 100644
--- a/package/urandom-scripts/S01seedrng
+++ b/package/urandom-scripts/S01seedrng
@@ -28,6 +28,9 @@ SEEDRNG_ARGS=""
 # shellcheck source=/dev/null
 [ -r "/etc/default/$DAEMON" ] && . "/etc/default/$DAEMON"
 
+# Reglinux
+mkdir -p /var/lib/seedrng
+
 case "$1" in
 	start|stop|restart|reload)
 		# Never fail, as this isn't worth making a fuss
