--- a/Makefile	2024-12-29 11:29:20.000000000 +0100
+++ b/Makefile	2025-01-22 16:53:28.529148745 +0100
@@ -23,6 +23,9 @@
 config.mak:
 endif
 
+# GCC 14 workaround
+CFLAGS += -Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types
+
 # This is actually needed, believe me - one bit is used as a flag in some tables
 # If you really have to disable this, set NO_ALIGN_FUNCTIONS elsewhere.
 ifndef NO_ALIGN_FUNCTIONS
