diff --git a/toolchain/Config.in b/toolchain/Config.in
index 94723da141..35fb1263a0 100644
--- a/toolchain/Config.in
+++ b/toolchain/Config.in
@@ -647,6 +647,26 @@ config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_12
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_11
 	select BR2_TOOLCHAIN_HEADERS_LATEST
 
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_13
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_12
+
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_14
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_13
+
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_15
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_14
+
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_16
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_15
+
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_17
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_16
+
 # This should be selected by the latest version, above, to indicate that
 # Buildroot does not know of more recent headers than the ones selected.
 # This allows using toolchains with headers more recent than Buildroot
@@ -658,6 +678,11 @@ config BR2_TOOLCHAIN_HEADERS_LATEST
 # stops affecting a value on the first matching default.
 config BR2_TOOLCHAIN_HEADERS_AT_LEAST
 	string
+	default "6.17" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_17
+	default "6.16" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_16
+	default "6.15" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_15
+	default "6.14" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_14
+	default "6.13" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_13
 	default "6.12" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_12
 	default "6.11" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_11
 	default "6.10" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_10
