diff --git a/package/strace/strace.hash b/package/strace/strace.hash
index 9ab5715e324..0eb2adce6d1 100644
--- a/package/strace/strace.hash
+++ b/package/strace/strace.hash
@@ -1,3 +1,4 @@
+sha256  0ad5dcba973a69e779650ef1cb335b12ee60716fc7326609895bd33e6d2a7325  strace-6.18.tar.xz
 # Locally calculated after checking signature with RSA key 0xA8041FA839E16E36
 # https://strace.io/files/6.13/strace-6.13.tar.xz.asc
 sha256  e209daf0ee038ca5adcc4c277e9273b4d51f46a2ff86da575d36742ac3508a17  strace-6.13.tar.xz
