diff --git a/package/strace/strace.hash b/package/strace/strace.hash
index 9ab5715e32..f860f194b2 100644
--- a/package/strace/strace.hash
+++ b/package/strace/strace.hash
@@ -1,3 +1,4 @@
+sha256  0a7c7bedc7efc076f3242a0310af2ae63c292a36dd4236f079e88a93e98cb9c0  strace-6.17.tar.xz
 # Locally calculated after checking signature with RSA key 0xA8041FA839E16E36
 # https://strace.io/files/6.13/strace-6.13.tar.xz.asc
 sha256  e209daf0ee038ca5adcc4c277e9273b4d51f46a2ff86da575d36742ac3508a17  strace-6.13.tar.xz
