From d087eb20c521943155612420c900b501f302194b Mon Sep 17 00:00:00 2001
From: sonninnos <tony@pihaus.org>
Date: Wed, 5 Feb 2025 23:36:56 +0200
Subject: [PATCH] Remove double optimizer flag

---
 scripts/toolchain.lua | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/scripts/toolchain.lua b/scripts/toolchain.lua
index 0e48376304040..7c42466433e8c 100644
--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -771,10 +771,6 @@ function toolchain(_buildDir, _subDir)
 		}
 	configuration { "libretro" }
 		targetdir (_buildDir .. "libretro" .. "/bin")
-		flags {
-			"Optimize",
-		}
-
 
 -- END   libretro overrides to MAME's GENie build
 
