diff --git a/package/libopenh264/Config.in b/package/libopenh264/Config.in
index 2b77578a832..79ad1b999d9 100644
--- a/package/libopenh264/Config.in
+++ b/package/libopenh264/Config.in
@@ -2,7 +2,7 @@ config BR2_PACKAGE_LIBOPENH264_ARCH_SUPPORTS
 	bool
 	default y if BR2_arm || BR2_armeb || BR2_aarch64 || BR2_i386 || \
 		BR2_mips || BR2_mipsel || BR2_mips64 || BR2_mips64el || \
-		BR2_x86_64
+		BR2_x86_64 || BR2_riscv
 
 config BR2_PACKAGE_LIBOPENH264
 	bool "libopenh264"
