--- a/Makefile	2025-03-09 16:25:16.658394714 +0100
+++ b/Makefile	2025-03-09 16:28:10.470657040 +0100
@@ -4,7 +4,7 @@
 RETRO_USE_HW_RENDER          ?= 1
 RETRO_DISABLE_PLUS           ?= 0
 
-CXXFLAGS_ALL = $(shell pkg-config --cflags --static sdl2 vorbisfile vorbis egl glesv1_cm glesv2) $(CXXFLAGS) \
+CXXFLAGS_ALL = $(shell pkg-config --cflags --static sdl2 vorbisfile vorbis egl glesv2) $(CXXFLAGS) \
         -DBASE_PATH='"$(BASE_PATH)"' \
         -IRSDKv4/               \
         -IRSDKv4/NativeObjects/ \
@@ -13,7 +13,7 @@
         -Idependencies/all/tinyxml2/
 
 LDFLAGS_ALL = $(LDFLAGS)
-LIBS_ALL = $(shell pkg-config --libs --static sdl2 vorbisfile vorbis egl glesv1_cm glesv2) -pthread $(LIBS)
+LIBS_ALL = $(shell pkg-config --libs --static vorbisfile vorbis egl glesv2) -pthread -lSDL2 $(LIBS)
 
 SOURCES = dependencies/all/tinyxml2/tinyxml2.cpp \
         RSDKv4/Animation.cpp     \
