diff --git a/package/x11r7/xlib_libX11/xlib_libX11.mk b/package/x11r7/xlib_libX11/xlib_libX11.mk
index 920a5d9aed3..07d3dd832dd 100644
--- a/package/x11r7/xlib_libX11/xlib_libX11.mk
+++ b/package/x11r7/xlib_libX11/xlib_libX11.mk
@@ -3,8 +3,8 @@
 # xlib_libX11
 #
 ################################################################################
-
-XLIB_LIBX11_VERSION = 1.8.7
+# reglinux - bump
+XLIB_LIBX11_VERSION = 1.8.9
 XLIB_LIBX11_SOURCE = libX11-$(XLIB_LIBX11_VERSION).tar.xz
 XLIB_LIBX11_SITE = https://xorg.freedesktop.org/archive/individual/lib
 XLIB_LIBX11_LICENSE = MIT
