--- a/scripts/amlogic/merge_pre_check.pl	2024-01-02 14:35:01.000000000 +0100
+++ b/scripts/amlogic/merge_pre_check.pl	2024-04-24 13:10:35.677838378 +0200
@@ -20,11 +20,8 @@
 sub check_defconfig
 {
 	my $arm64_config = `git diff  --name-only arch/arm64/configs | grep  -E *defconfig`;
-	my $is_common_drivers = `pwd | grep -E *common_drivers*`;
-	if (!$arm64_config || !$is_common_drivers)
-	{
-		return 0;
-	}
+	return 0;
+
 	my $err = 0;
 	#print "2. Check meson_defconfig: ";
 	if( -e $config1)
