--- a/drivers/amlogic/drm/meson_hdmi.c	2024-05-23 22:15:52.874008021 +0200
+++ b/drivers/amlogic/drm/meson_hdmi.c	2024-05-23 22:19:45.246990079 +0200
@@ -826,8 +826,8 @@
 	struct am_hdmitx_connector_state *new_hdmitx_state, *old_hdmitx_state;
 	struct drm_crtc_state *new_crtc_state = NULL;
 	struct am_hdmi_tx *am_hdmitx = connector_to_am_hdmi(connector);
-	unsigned int hdmitx_content_type =
-		am_hdmitx->hdmitx_dev->get_content_types();
+	/*unsigned int hdmitx_content_type =
+		am_hdmitx->hdmitx_dev->get_content_types();*/
 
 	if (!state) {
 		DRM_ERROR("state is NULL.\n");
@@ -940,10 +940,10 @@
 void meson_hdmitx_atomic_print_state(struct drm_printer *p,
 	const struct drm_connector_state *state)
 {
-	struct am_hdmitx_connector_state *hdmitx_state =
+	/*struct am_hdmitx_connector_state *hdmitx_state =
 		to_am_hdmitx_connector_state(state);
 
-	/*drm_printf(p, "\tdrm hdmitx state:\n");
+	drm_printf(p, "\tdrm hdmitx state:\n");
 	drm_printf(p, "\t\t VRR_CAP:[%u]\n", am_hdmi_info.hdmitx_dev->get_vrr_cap ?
 		am_hdmi_info.hdmitx_dev->get_vrr_cap() : 0);
 	drm_printf(p, "\t\t android_path:[%d]\n", am_hdmi_info.android_path);
@@ -955,10 +955,10 @@
 	drm_printf(p, "\t\t picture_aspect_ratio:[%d]\n", state->picture_aspect_ratio);*/
 }
 
+#if 0
 static bool meson_hdmitx_is_hdcp_running(void)
 {
 	return false;
-#if 0
 	if (am_hdmi_info.hdcp_state == HDCP_STATE_DISCONNECT ||
 		am_hdmi_info.hdcp_state == HDCP_STATE_STOP)
 		return false;
@@ -968,8 +968,8 @@
 			am_hdmi_info.hdcp_state);
 
 	return true;
-#endif
 }
+#endif
 
 static void meson_hdmitx_set_hdcp_result(int result)
 {
@@ -1005,9 +1005,9 @@
 #endif
 }
 
+#if 0
 static void meson_hdmitx_start_hdcp(int hdcp_mode)
 {
-#if 0
 	if (hdcp_mode == HDCP_NULL)
 		return;
 
@@ -1016,38 +1016,38 @@
 	am_hdmi_info.hdmitx_dev->hdcp_enable(hdcp_mode);
 	DRM_DEBUG("start hdcp [%d-%d]...\n",
 		am_hdmi_info.hdcp_request_content_type, am_hdmi_info.hdcp_mode);
-#endif
 }
+#endif
 
 static void meson_hdmitx_stop_hdcp(void)
 {
+	meson_hdmitx_set_hdcp_result(HDCP_AUTH_UNKNOWN);
 #if 0
 	if (meson_hdmitx_is_hdcp_running()) {
 		am_hdmi_info.hdmitx_dev->hdcp_disable();
 		am_hdmi_info.hdcp_state = HDCP_STATE_STOP;
 		am_hdmi_info.hdcp_mode = HDCP_NULL;
-		meson_hdmitx_set_hdcp_result(HDCP_AUTH_UNKNOWN);
 	}
 #endif
 }
 
 static void meson_hdmitx_disconnect_hdcp(void)
 {
+	meson_hdmitx_set_hdcp_result(HDCP_AUTH_UNKNOWN);
 #if 0
 	if (meson_hdmitx_is_hdcp_running()) {
 		am_hdmi_info.hdmitx_dev->hdcp_disconnect();
 		am_hdmi_info.hdcp_state = HDCP_STATE_DISCONNECT;
 		am_hdmi_info.hdcp_mode = HDCP_NULL;
-		meson_hdmitx_set_hdcp_result(HDCP_AUTH_UNKNOWN);
 	}
 #endif
 }
 
+#if 0
 static int meson_hdmitx_get_hdcp_request(struct am_hdmi_tx *tx,
 	int request_type_mask)
 {
 	return HDCP_NULL;
-#if 0
 	struct meson_hdmitx_dev *tx_dev = tx->hdmitx_dev;
 	int type;
 	unsigned int hdcp_tx_type = tx_dev->get_tx_hdcp_cap();
@@ -1084,11 +1084,12 @@
 		break;
 	}
 	return type;
-#endif
 }
+#endif
 
 void meson_hdmitx_update_hdcp(void)
 {
+	meson_hdmitx_set_hdcp_result(HDCP_AUTH_OK);
 #if 0
 	int hdcp_request_mode = HDCP_NULL;
 	int hdcp_request_mask = HDCP_NULL;
@@ -1178,6 +1179,7 @@
 	struct drm_modeset_lock *mode_lock =
 		&connector->dev->mode_config.connection_mutex;
 	drm_modeset_unlock(mode_lock);
+	meson_hdmitx_set_hdcp_result(HDCP_AUTH_OK);
 #if 0
 	bool locked_outer = drm_modeset_is_locked(mode_lock);
 
