From 74c58b8e746fc15d033980a7612c2f76861d4db6 Mon Sep 17 00:00:00 2001
From: "xiaojin.cxj" <xiaojin.cxj@linux.alibaba.com>
Date: Fri, 7 Jun 2024 15:24:00 +0800
Subject: [PATCH 123/400] fix rpmsg addr cast warning

---
 drivers/rpmsg/th1520_rpmsg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/rpmsg/th1520_rpmsg.c b/drivers/rpmsg/th1520_rpmsg.c
index 8964ecb54524..abc8338ec497 100644
--- a/drivers/rpmsg/th1520_rpmsg.c
+++ b/drivers/rpmsg/th1520_rpmsg.c
@@ -165,7 +165,7 @@ static bool th1520_rpmsg_notify(struct virtqueue *vq)
 static int th1520_mu_rpmsg_callback(struct notifier_block *this,
 				    unsigned long index, void *data)
 {
-	phys_addr_t mu_msg = (phys_addr_t)data;
+	uintptr_t mu_msg = (uintptr_t)data;
 	struct th1520_virdev *virdev;
 
 	virdev = container_of(this, struct th1520_virdev, nb);
@@ -414,7 +414,7 @@ static int set_vring_phy_buf(struct platform_device *pdev,
 
 static void rpmsg_work_handler(struct work_struct *work)
 {
-	phys_addr_t message = 0;
+	uintptr_t message = 0;
 	struct delayed_work *dwork = to_delayed_work(work);
 	struct th1520_rpmsg_vproc *rpdev =
 		container_of(dwork, struct th1520_rpmsg_vproc, rpmsg_work);
-- 
2.43.0

