--- a/backends/platform/libretro/dependencies.mk	2026-02-23 23:40:24.461572540 +0100
+++ b/backends/platform/libretro/dependencies.mk	2026-02-23 23:40:53.940052144 +0100
@@ -7,7 +7,7 @@
 
 DEPS_FOLDER_libretro-deps   := libretro-deps
 DEPS_URL_libretro-deps      := https://github.com/libretro/libretro-deps
-DEPS_COMMIT_libretro-deps   := abf5246b016569759e7d1b0ea91bb98c2e34d160
+DEPS_COMMIT_libretro-deps   := a2a865e74f58035c416a0c63be5772fdcf35b071
 
 DEPS_FOLDER_libretro-common := libretro-common
 DEPS_URL_libretro-common    := https://github.com/libretro/libretro-common
