diff --git a/package/linux-headers/Config.in.host b/package/linux-headers/Config.in.host
index 4084a36436..da6a7952c5 100644
--- a/package/linux-headers/Config.in.host
+++ b/package/linux-headers/Config.in.host
@@ -49,6 +49,18 @@ config BR2_KERNEL_HEADERS_6_12
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_12
 	select BR2_KERNEL_HEADERS_LATEST
 
+config BR2_KERNEL_HEADERS_6_13
+	bool "Linux 6.13.x kernel headers"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_13
+
+config BR2_KERNEL_HEADERS_6_15
+	bool "Linux 6.15.x kernel headers"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_15
+
+config BR2_KERNEL_HEADERS_6_16
+	bool "Linux 6.15.x kernel headers"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_16
+
 config BR2_KERNEL_HEADERS_VERSION
 	bool "Manually specified Linux version"
 	help
@@ -125,6 +137,22 @@ choice
 	  If your kernel headers are more recent than the latest version
 	  in the choice, then select the latest version.
 
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_6_16
+	bool "6.16.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_16
+
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_6_15
+	bool "6.15.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_15
+
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_6_14
+	bool "6.14.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_14
+
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_6_13
+	bool "6.13.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_13
+
 config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_6_12
 	bool "6.12.x or later"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_6_12
@@ -433,7 +461,9 @@ config BR2_DEFAULT_KERNEL_HEADERS
 	default "5.15.193"	if BR2_KERNEL_HEADERS_5_15
 	default "6.1.152"	if BR2_KERNEL_HEADERS_6_1
 	default "6.6.106"	if BR2_KERNEL_HEADERS_6_6
-	default "6.12.47"	if BR2_KERNEL_HEADERS_6_12
+	default "6.12.41"	if BR2_KERNEL_HEADERS_6_12
+	default "6.15.11"	if BR2_KERNEL_HEADERS_6_15
+	default "6.16.4"	if BR2_KERNEL_HEADERS_6_16
 	default BR2_DEFAULT_KERNEL_VERSION if BR2_KERNEL_HEADERS_VERSION
 	default "custom"	if BR2_KERNEL_HEADERS_CUSTOM_TARBALL
 	default BR2_KERNEL_HEADERS_CUSTOM_REPO_VERSION \
