diff --git a/package/libfreeimage/libfreeimage.mk b/package/libfreeimage/libfreeimage.mk
index 4775f92109a..b0003c954bb 100644
--- a/package/libfreeimage/libfreeimage.mk
+++ b/package/libfreeimage/libfreeimage.mk
@@ -19,8 +19,22 @@ define LIBFREEIMAGE_EXTRACT_CMDS
 	rmdir $(@D)/FreeImage
 endef
 
+# batocera
+ifeq ($(BR2_ARM_CPU_HAS_NEON),y)
+	LIBFREEIMAGE_CFLAGS=$(TARGET_CFLAGS) -DPNG_ARM_NEON_OPT=0
+endif
+ifeq ($(BR2_ARM_FPU_FP_ARMV8),y)
+	LIBFREEIMAGE_CFLAGS=$(TARGET_CFLAGS) -DPNG_ARM_NEON_OPT=0
+endif
+ifeq ($(BR2_ARM_CPU_ARMV8A),y)
+	LIBFREEIMAGE_CFLAGS=$(TARGET_CFLAGS) -DPNG_ARM_NEON_OPT=0
+endif
+
+# REG fix GCC 14
+LIBFREEIMAGE_CFLAGS += -Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types
+
 define LIBFREEIMAGE_BUILD_CMDS
-	$(TARGET_MAKE_ENV) $(TARGET_CONFIGURE_OPTS) \
+	$(TARGET_MAKE_ENV) $(TARGET_CONFIGURE_OPTS) CFLAGS="$(LIBFREEIMAGE_CFLAGS)" \
 		CXXFLAGS="$(TARGET_CXXFLAGS) -std=c++11" $(MAKE) -C $(@D)
 endef
 
