--- a/hid-lg.c	2024-11-18 20:21:23.046659863 +0100
+++ b/hid-lg.c	2024-11-18 20:21:31.624825633 +0100
@@ -427,7 +427,7 @@
  * above the logical maximum described in descriptor. This extends
  * the original value of 0x28c of logical maximum to 0x104d
  */
-static __u8 *lg_report_fixup(struct hid_device *hdev, __u8 *rdesc,
+static const __u8 *lg_report_fixup(struct hid_device *hdev, __u8 *rdesc,
 		unsigned int *rsize)
 {
 	struct lg_drv_data *drv_data = hid_get_drvdata(hdev);
