--- a/nuitka/utils/ReExecute.py	2024-04-19 15:31:38.000000000 +0200
+++ b/nuitka/utils/ReExecute.py	2024-05-01 13:39:07.499435936 +0200
@@ -48,13 +48,14 @@
             os._exit(2)
     else:
         # The star arguments is the API of execl
+        print(*args)
         os.execl(*args)
 
 
 def reExecuteNuitka(pgo_filename):
     # Execute with full path as the process name, so it can find itself and its
     # libraries.
-    args = [sys.executable, sys.executable]
+    args = [os.environ["QEMU_USER"], sys.executable, sys.executable]
 
     if sys.version_info >= (3, 7) and sys.flags.utf8_mode:
         args += ["-X", "utf8"]
--- a/nuitka/utils/Execution.py	2024-04-19 15:31:38.000000000 +0200
+++ b/nuitka/utils/Execution.py	2024-05-01 13:39:07.499435936 +0200
@@ -420,6 +420,11 @@
 def executeProcess(
     command, env=None, stdin=False, shell=False, external_cwd=False, timeout=None
 ):
+    runner = os.environ["QEMU_USER"];
+    command = list(command)
+    command.insert(0, runner)
+    command = tuple(command)
+    print(command)
     process = createProcess(
         command=command, env=env, stdin=stdin, shell=shell, external_cwd=external_cwd
     )
--- a/nuitka/build/DataComposerInterface.py	2024-04-19 15:31:38.000000000 +0200
+++ b/nuitka/build/DataComposerInterface.py	2024-05-01 13:39:07.499435936 +0200
@@ -60,6 +60,7 @@
         try:
             subprocess.check_call(
                 [
+                    os.environ["QEMU_USER"],
                     sys.executable,
                     data_composer_path,
                     source_dir,
