diff -u -r /home/romain/dev/REG-Linux/output/cha/build/nuitka-2.1.6/nuitka/build/SconsInterface.py ./build/SconsInterface.py
--- a/nuitka/build/SconsInterface.py	2024-04-19 15:31:38.000000000 +0200
+++ b/nuitka/build/SconsInterface.py	2024-05-02 00:27:50.227135127 +0200
@@ -275,7 +275,8 @@
         env_values["_NUITKA_BUILD_DEFINITIONS_CATALOG"] = ",".join(env_values.keys())
 
         # Pass quiet setting to scons via environment variable.
-        env_values["NUITKA_QUIET"] = "1" if Tracing.is_quiet else "0"
+        #env_values["NUITKA_QUIET"] = "1" if Tracing.is_quiet else "0"
+        env_values["NUITKA_QUIET"] = "0"
 
         scons_command = _buildSconsCommand(
             options=options, scons_filename=scons_filename
@@ -286,6 +287,8 @@
 
         Tracing.flushStandardOutputs()
 
+        print(scons_command)
+
         with withEnvironmentVarsOverridden(env_values):
             try:
                 result = subprocess.call(scons_command, shell=False, cwd=source_dir)
diff -u -r /home/romain/dev/REG-Linux/output/cha/build/nuitka-2.1.6/nuitka/utils/Execution.py ./utils/Execution.py
--- a/nuitka/utils/Execution.py	2024-05-01 16:30:22.251716422 +0200
+++ b/nuitka/utils/Execution.py	2024-05-02 00:27:50.239135411 +0200
@@ -348,7 +348,12 @@
 
     # Allow to avoid repeated scans in PATH for the tool.
     command[0] = getExecutablePath(tool)
-
+    is_target_patchelf = "target/usr/bin/patchelf" in command[0]
+    is_staging_patchelf = "sysroot/usr/bin/patchelf" in command[0]
+    # /cha/host/arm-buildroot-linux-gnueabihf/sysroot/usr/bin/patchelf
+    if is_target_patchelf or is_staging_patchelf:
+        command.insert(0, os.environ["QEMU_USER"])
+    print("executeToolChecked " , command)
     process = subprocess.Popen(
         command,
         stdin=getNullInput(),
@@ -422,7 +427,11 @@
 ):
     runner = os.environ["QEMU_USER"];
     command = list(command)
-    command.insert(0, runner)
+    print(command[0])
+    is_target_python =  "target/usr/bin/python" in command[0]
+    print("Result =" + str(is_target_python))
+    if is_target_python is True:
+        command.insert(0, runner)
     command = tuple(command)
     print(command)
     process = createProcess(
