--- a/ruby/cmake/os-linux.cmake	2025-01-25 21:14:42.000000000 +0100
+++ b/ruby/cmake/os-linux.cmake	2025-01-26 01:18:20.337752533 +0100
@@ -132,7 +132,7 @@
   PRIVATE
     $<$<BOOL:${SDL_FOUND}>:SDL::SDL>
     $<$<BOOL:${OpenAL_FOUND}>:OpenAL::OpenAL>
-    $<$<BOOL:TRUE>:librashader::librashader>
+    $<$<BOOL:${librashader_FOUND}>:librashader::librashader>
     $<$<BOOL:${OSS_FOUND}>:OSS::OSS>
     $<$<BOOL:${ALSA_FOUND}>:ALSA::ALSA>
     $<$<BOOL:${PulseAudio_FOUND}>:PulseAudio::PulseAudio>
--- a/cmake/linux/compilerconfig.cmake	2025-02-05 22:44:51.445267932 +0100
+++ b/cmake/linux/compilerconfig.cmake	2025-02-05 22:45:10.110697433 +0100
@@ -28,13 +28,3 @@
   add_compile_options(${_ares_gcc_common_options})
 endif()
 
-if(ARES_BUILD_LOCAL)
-  add_compile_options($<$<NOT:$<CONFIG:Debug>>:-march=native>)
-else()
-  if(ARES_ENABLE_MINIMUM_CPU)
-    string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} LOWERCASE_CMAKE_SYSTEM_PROCESSOR)
-    if(LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|x86|x64|x86_64|amd64|e2k)")
-      add_compile_options($<$<NOT:$<CONFIG:Debug>>:-march=x86-64-v2>)
-    endif()
-  endif()
-endif()
