--- a/ruby/video/opengl/opengl.hpp	2025-02-05 14:28:51.000000000 +0100
+++ b/ruby/video/opengl/opengl.hpp	2025-02-05 22:53:44.102428545 +0100
@@ -16,11 +16,11 @@
   #error "ruby::OpenGL3: unsupported platform"
 #endif
 
-#if defined(CMAKE)
+/*#if defined(CMAKE)
 #include <librashader/librashader_ld.h>
 #else
 #include "librashader_ld.h"
-#endif
+#endif*/
 
 #include "bind.hpp"
 #include "utility.hpp"
