--- a/src/common/fs/path_util.cpp	2024-10-07 16:03:01.542485766 +0200
+++ b/src/common/fs/path_util.cpp	2024-10-07 16:04:13.353187220 +0200
@@ -155,13 +155,13 @@
         GenerateEdenPath(EdenPath::ConfigDir, eden_path_config);
         GenerateEdenPath(EdenPath::CrashDumpsDir, eden_path / CRASH_DUMPS_DIR);
         GenerateEdenPath(EdenPath::DumpDir, eden_path / DUMP_DIR);
-        GenerateEdenPath(EdenPath::KeysDir, eden_path / KEYS_DIR);
+        GenerateEdenPath(EdenPath::KeysDir, "/userdata/bios/switch");
         GenerateEdenPath(EdenPath::LoadDir, eden_path / LOAD_DIR);
-        GenerateEdenPath(EdenPath::LogDir, eden_path / LOG_DIR);
+        GenerateEdenPath(EdenPath::LogDir, "/userdata/system/logs");
         GenerateEdenPath(EdenPath::NANDDir, eden_path / NAND_DIR);
         GenerateEdenPath(EdenPath::PlayTimeDir, eden_path / PLAY_TIME_DIR);
         GenerateEdenPath(EdenPath::SaveDir, eden_path / NAND_DIR);
-        GenerateEdenPath(EdenPath::ScreenshotsDir, eden_path / SCREENSHOTS_DIR);
+        GenerateEdenPath(EdenPath::ScreenshotsDir, "/userdata/screenshots/switch/eden");
         GenerateEdenPath(EdenPath::SDMCDir, eden_path / SDMC_DIR);
         GenerateEdenPath(EdenPath::ShaderDir, eden_path / SHADER_DIR);
         GenerateEdenPath(EdenPath::TASDir, eden_path / TAS_DIR);
