diff --git a/package/x11r7/xwayland/xwayland.mk b/package/x11r7/xwayland/xwayland.mk
index e3b6727c19..08927f86a6 100644
--- a/package/x11r7/xwayland/xwayland.mk
+++ b/package/x11r7/xwayland/xwayland.mk
@@ -30,6 +30,14 @@ XWAYLAND_CONF_OPTS = \
 	-Ddtrace=false \
 	-Ddocs=false
 
+# reglinux add mesa3d or img-mesa3d (PowerVR)
+ifeq ($(BR2_PACKAGE_MESA3D),y)
+XWAYLAND_DEPENDENCIES += mesa3d
+endif
+ifeq ($(BR2_PACKAGE_IMG_MESA3D),y)
+XWAYLAND_DEPENDENCIES += img-mesa3d
+endif
+
 ifeq ($(BR2_PACKAGE_LIBEPOXY),y)
 XWAYLAND_CONF_OPTS += -Dglamor=true
 XWAYLAND_DEPENDENCIES += libepoxy
