diff --git a/package/wireplumber/wireplumber.hash b/package/wireplumber/wireplumber.hash
index 7fff34a7a74..550d64b276e 100644
--- a/package/wireplumber/wireplumber.hash
+++ b/package/wireplumber/wireplumber.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  4926399cffe3da12f1338e1feec53ef944e86ae7c719384bff87490d7e3edbc7  wireplumber-0.5.5.tar.bz2
+sha256  7782ba07ad028e77e8a1c4b8d5e702a63468cfad6ddaa3a1e683daa8989450ab  wireplumber-0.5.7.tar.bz2
 sha256  ff2eb9af5b05d92ac3a38cffcad27582ef555a31ff1781e1c1d4e087f77c9c81  LICENSE
