diff -u -r a/daemon/gamemode-context.c b/daemon/gamemode-context.c
--- a/daemon/gamemode-context.c	2024-08-19 12:18:51.000000000 +0200
+++ b/daemon/gamemode-context.c	2025-03-18 23:34:09.154473905 +0100
@@ -49,7 +49,9 @@
 #include <stdatomic.h>
 #include <stdlib.h>
 #include <sys/time.h>
-#include <systemd/sd-daemon.h> /* TODO: Move usage to gamemode-dbus.c */
+#ifndef USE_BASU
+#include <systemd/sd-daemon.h> /* TODO: Move usage to gamemode-dbus.c */
+#endif
 #include <unistd.h>
 
 /**
@@ -415,7 +415,9 @@
 static void game_mode_context_enter(GameModeContext *self)
 {
 	LOG_MSG("Entering Game Mode...\n");
-	sd_notifyf(0, "STATUS=%sGameMode is now active.%s\n", "\x1B[1;32m", "\x1B[0m");
+#ifndef USE_BASU
+	sd_notifyf(0, "STATUS=%sGameMode is now active.%s\n", "\x1B[1;32m", "\x1B[0m");
+#endif
 
 	if (game_mode_set_governor(self, GAME_MODE_GOVERNOR_DESIRED) == 0) {
 		/* We just switched to a non-default governor.  Enable the iGPU
@@ -456,7 +456,9 @@
 static void game_mode_context_leave(GameModeContext *self)
 {
 	LOG_MSG("Leaving Game Mode...\n");
-	sd_notifyf(0, "STATUS=%sGameMode is currently deactivated.%s\n", "\x1B[1;36m", "\x1B[0m");
+#ifndef USE_BASU
+	sd_notifyf(0, "STATUS=%sGameMode is currently deactivated.%s\n", "\x1B[1;36m", "\x1B[0m");
+#endif
 
 	/* Remove GPU optimisations */
 	game_mode_apply_gpu(self->stored_gpu);
diff -u -r a/daemon/gamemode-dbus.c b/daemon/gamemode-dbus.c
--- a/daemon/gamemode-dbus.c	2024-08-19 12:18:51.000000000 +0200
+++ b/daemon/gamemode-dbus.c	2025-03-18 23:33:28.504497568 +0100
@@ -36,9 +36,11 @@
 #include "common-logging.h"
 #include "common-pidfds.h"
 
-#ifdef USE_ELOGIND
+#if defined(USE_ELOGIND)
 #include <elogind/sd-bus.h>
 #include <elogind/sd-daemon.h>
+#elif defined(USE_BASU)
+#include <basu/sd-bus.h>
 #else
 #include <systemd/sd-bus.h>
 #include <systemd/sd-daemon.h>
@@ -686,7 +688,9 @@
 	}
 
 	LOG_MSG("Successfully initialised bus with name [%s]...\n", "com.feralinteractive.GameMode");
-	sd_notifyf(0, "STATUS=%sGameMode is ready to be activated.%s\n", "\x1B[1;36m", "\x1B[0m");
+#ifndef USE_BASU
+	sd_notifyf(0, "STATUS=%sGameMode is ready to be activated.%s\n", "\x1B[1;36m", "\x1B[0m");
+#endif
 
 	/* Now loop, waiting for callbacks */
 	for (;;) {
diff -u -r a/daemon/gamemoded.c b/daemon/gamemoded.c
--- a/daemon/gamemoded.c	2024-08-19 12:18:51.000000000 +0200
+++ b/daemon/gamemoded.c	2025-03-18 23:34:17.484673622 +0100
@@ -61,7 +61,9 @@
 #include <getopt.h>
 #include <signal.h>
 #include <sys/stat.h>
-#include <systemd/sd-daemon.h> /* TODO: Move usage to gamemode-dbus.c */
+#ifndef USE_BASU
+#include <systemd/sd-daemon.h> /* TODO: Move usage to gamemode-dbus.c */
+#endif
 #include <unistd.h>
 
 #define USAGE_TEXT                                                                                 \
@@ -83,7 +83,9 @@
 static void sigint_handler(__attribute__((unused)) int signo)
 {
 	LOG_MSG("Quitting by request...\n");
-	sd_notify(0, "STATUS=GameMode is quitting by request...\n");
+#ifndef USE_BASU
+	sd_notify(0, "STATUS=GameMode is quitting by request...\n");
+#endif
 
 	/* Clean up nicely */
 	game_mode_context_destroy(game_mode_context_instance());
@@ -337,5 +337,7 @@
 
 	/* Log we're finished */
 	LOG_MSG("Quitting naturally...\n");
-	sd_notify(0, "STATUS=GameMode is quitting naturally...\n");
+#ifndef USE_BASU
+	sd_notify(0, "STATUS=GameMode is quitting naturally...\n");
+#endif
 }
diff -u -r a/meson.build b/meson.build
--- a/meson.build	2024-08-19 12:18:51.000000000 +0200
+++ b/meson.build	2025-03-18 23:25:21.601699355 +0100
@@ -91,6 +91,9 @@
 elif sd_bus_provider == 'elogind'
     sd_bus_args += ['-DUSE_ELOGIND']
     sd_bus_dep = dependency('libelogind')
+elif sd_bus_provider == 'basu'
+    sd_bus_args += ['-DUSE_BASU']
+    sd_bus_dep = dependency('basu')
 endif
 
 # For the client, libdbus is used
diff -u -r a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2024-08-19 12:18:51.000000000 +0200
+++ b/meson_options.txt	2025-03-18 23:22:36.581937522 +0100
@@ -3,7 +3,7 @@
 option('with-pam-limits-dir', type: 'string', description: 'Explicitly set the PAM limits.d directory', value: '/etc/security/limits.d')
 
 # sd-bus provider
-option('with-sd-bus-provider', type: 'combo', choices: ['systemd', 'elogind', 'no-daemon'], value: 'systemd')
+option('with-sd-bus-provider', type: 'combo', choices: ['systemd', 'elogind', 'basu', 'no-daemon'], value: 'systemd')
 
 # systemd specific
 option('with-systemd-user-unit', type: 'boolean', description: 'Install systemd user unit', value: 'true')
