diff --git a/package/valgrind/Config.in b/package/valgrind/Config.in
index 56fda07d092..43d33bb90a8 100644
--- a/package/valgrind/Config.in
+++ b/package/valgrind/Config.in
@@ -1,6 +1,6 @@
 config BR2_PACKAGE_VALGRIND_ARCH_SUPPORTS
 	bool
-	default y if BR2_aarch64
+	default y if BR2_arm || BR2_aarch64
 	default y if BR2_ARM_CPU_ARMV7A
 	default y if BR2_mips || BR2_mipsel || BR2_mips64 || BR2_mips64el
 	default y if BR2_i386 || BR2_x86_64
