diff --git a/package/x11r7/xlib_libXdmcp/xlib_libXdmcp.mk b/package/x11r7/xlib_libXdmcp/xlib_libXdmcp.mk
index a79e2615dba..0100fb45c4f 100644
--- a/package/x11r7/xlib_libXdmcp/xlib_libXdmcp.mk
+++ b/package/x11r7/xlib_libXdmcp/xlib_libXdmcp.mk
@@ -3,8 +3,8 @@
 # xlib_libXdmcp
 #
 ################################################################################
-
-XLIB_LIBXDMCP_VERSION = 1.1.4
+# reglinux - bump
+XLIB_LIBXDMCP_VERSION = 1.1.5
 XLIB_LIBXDMCP_SOURCE = libXdmcp-$(XLIB_LIBXDMCP_VERSION).tar.xz
 XLIB_LIBXDMCP_SITE = https://xorg.freedesktop.org/archive/individual/lib
 XLIB_LIBXDMCP_LICENSE = MIT
