--- a/drivers/gpu/drm/drm_panel_orientation_quirks.c	2024-07-25 14:30:32.424290849 +0200
+++ b/drivers/gpu/drm/drm_panel_orientation_quirks.c	2024-07-25 14:31:45.389928268 +0200
@@ -208,6 +208,12 @@
 		  DMI_MATCH(DMI_BOARD_NAME, "KUN"),
 		},
 		.driver_data = (void *)&lcd1600x2560_rightside_up,
+	}, {	/* AYN LOKI ZERO */
+		.matches = {
+		  DMI_EXACT_MATCH(DMI_BOARD_VENDOR, "ayn"),
+		  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Loki Zero"),
+		},
+		.driver_data = (void *)&lcd1080x1920_leftside_up,
 	}, {	/* Chuwi HiBook (CWI514) */
 		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "Hampoo"),
