--- a/rpcs3/rpcs3qt/gs_frame.cpp	2025-02-28 17:57:08.190323559 +0100
+++ b/rpcs3/rpcs3qt/gs_frame.cpp	2025-02-28 17:58:02.785554194 +0100
@@ -36,10 +36,10 @@
 #elif defined(__APPLE__)
 //nothing
 #else
-#ifdef HAVE_WAYLAND
+/*#ifdef HAVE_WAYLAND
 #include <QGuiApplication>
 #include <qpa/qplatformnativeinterface.h>
-#endif
+#endif*/
 #ifdef HAVE_X11
 #include <X11/Xlib.h>
 #endif
@@ -662,7 +662,7 @@
 #elif defined(__APPLE__)
 	return reinterpret_cast<void*>(this->winId()); //NSView
 #else
-#ifdef HAVE_WAYLAND
+/*#ifdef HAVE_WAYLAND
 	QPlatformNativeInterface *native = QGuiApplication::platformNativeInterface();
 	struct wl_display *wl_dpy = static_cast<struct wl_display *>(
 		native->nativeResourceForWindow("display", NULL));
@@ -673,7 +673,7 @@
 		return std::make_pair(wl_dpy, wl_surf);
 	}
 	else
-#endif
+#endif*/
 #ifdef HAVE_X11
 	{
 		return std::make_pair(XOpenDisplay(0), static_cast<ulong>(this->winId()));
