--- a/arch/arm/mach-rockchip/rk3588/rk3588.c	2024-12-09 23:09:28.000000000 +0100
+++ b/arch/arm/mach-rockchip/rk3588/rk3588.c	2024-12-18 19:10:44.518054640 +0100
@@ -4,6 +4,11 @@
  * Copyright (c) 2022 Edgeble AI Technologies Pvt. Ltd.
  */
 
+#define LOG_CATEGORY LOGC_ARCH
+
+#include <dm.h>
+#include <fdt_support.h>
+#include <misc.h>
 #include <spl.h>
 #include <asm/armv8/mmu.h>
 #include <asm/arch-rockchip/bootrom.h>
@@ -177,4 +182,180 @@
 
 	return 0;
 }
+#endif
+
+#define RK3588_OTP_CPU_CODE_OFFSET		0x02
+#define RK3588_OTP_SPECIFICATION_OFFSET		0x06
+#define RK3588_OTP_IP_STATE_OFFSET		0x1d
+
+#define BAD_CPU_CLUSTER0		GENMASK(3, 0)
+#define BAD_CPU_CLUSTER1		GENMASK(5, 4)
+#define BAD_CPU_CLUSTER2		GENMASK(7, 6)
+#define BAD_GPU				GENMASK(4, 1)
+
+int checkboard(void)
+{
+	u8 cpu_code[2], specification, package;
+	struct udevice *dev;
+	char suffix[3];
+	int ret;
+
+	if (!IS_ENABLED(CONFIG_ROCKCHIP_OTP) || !CONFIG_IS_ENABLED(MISC))
+		return 0;
+
+	ret = uclass_get_device_by_driver(UCLASS_MISC,
+					  DM_DRIVER_GET(rockchip_otp), &dev);
+	if (ret) {
+		log_debug("Could not find otp device, ret=%d\n", ret);
+		return 0;
+	}
+
+	/* cpu-code: SoC model, e.g. 0x35 0x82 or 0x35 0x88 */
+	ret = misc_read(dev, RK3588_OTP_CPU_CODE_OFFSET, cpu_code, 2);
+	if (ret < 0) {
+		log_debug("Could not read cpu-code, ret=%d\n", ret);
+		return 0;
+	}
+
+	/* specification: SoC variant, e.g. 0xA for RK3588J and 0x13 for RK3588S */
+	ret = misc_read(dev, RK3588_OTP_SPECIFICATION_OFFSET, &specification, 1);
+	if (ret < 0) {
+		log_debug("Could not read specification, ret=%d\n", ret);
+		return 0;
+	}
+	/* package: likely SoC variant revision, 0x2 for RK3588S2 */
+	package = specification >> 5;
+	specification &= 0x1f;
+
+	/* for RK3588J i.e. '@' + 0xA = 'J' */
+	suffix[0] = specification > 1 ? '@' + specification : '\0';
+	/* for RK3588S2 i.e. '0' + 0x2 = '2' */
+	suffix[1] = package > 1 ? '0' + package : '\0';
+	suffix[2] = '\0';
+
+	printf("SoC:   RK%02x%02x%s\n", cpu_code[0], cpu_code[1], suffix);
+
+	return 0;
+}
+
+#ifdef CONFIG_OF_SYSTEM_SETUP
+static void fdt_path_del_node(void *fdt, const char *path)
+{
+	int node;
+
+	node = fdt_path_offset(fdt, path);
+	if (node >= 0)
+		fdt_del_node(fdt, node);
+}
+
+static void fdt_path_set_name(void *fdt, const char *path, const char *name)
+{
+	int node;
+
+	node = fdt_path_offset(fdt, path);
+	if (node >= 0)
+		fdt_set_name(fdt, node, name);
+}
+
+int ft_system_setup(void *blob, struct bd_info *bd)
+{
+	const char *cpu_node_names[] = {
+		"cpu@0", "cpu@100", "cpu@200", "cpu@300",
+		"cpu@400", "cpu@500", "cpu@600", "cpu@700",
+	};
+	struct udevice *dev;
+	u8 cpu_code[2], ip_state[3];
+	int parent, node, i, ret;
+
+	if (!IS_ENABLED(CONFIG_ROCKCHIP_OTP) || !CONFIG_IS_ENABLED(MISC))
+		return -ENOSYS;
+
+	ret = uclass_get_device_by_driver(UCLASS_MISC,
+					  DM_DRIVER_GET(rockchip_otp), &dev);
+	if (ret) {
+		log_debug("Could not find otp device, ret=%d\n", ret);
+		return ret;
+	}
+
+	/* cpu-code: SoC model, e.g. 0x35 0x82 or 0x35 0x88 */
+	ret = misc_read(dev, RK3588_OTP_CPU_CODE_OFFSET, cpu_code, 2);
+	if (ret < 0) {
+		log_debug("Could not read cpu-code, ret=%d\n", ret);
+		return ret;
+	}
+
+	log_debug("cpu-code: %02x %02x\n", cpu_code[0], cpu_code[1]);
+
+	/* skip on rk3588 */
+	if (cpu_code[0] == 0x35 && cpu_code[1] == 0x88)
+		return 0;
+
+	ret = misc_read(dev, RK3588_OTP_IP_STATE_OFFSET, &ip_state, 3);
+	if (ret < 0) {
+		log_debug("Could not read ip-state, ret=%d\n", ret);
+		return ret;
+	}
+
+	log_debug("ip-state: %02x %02x %02x\n",
+		  ip_state[0], ip_state[1], ip_state[2]);
+
+	if (cpu_code[0] == 0x35 && cpu_code[1] == 0x82) {
+		/* policy: always disable gpu */
+		ip_state[1] |= BAD_GPU;
+
+		/* policy: always disable one big core cluster */
+		if (!(ip_state[0] & (BAD_CPU_CLUSTER1|BAD_CPU_CLUSTER2)))
+			ip_state[0] |= BAD_CPU_CLUSTER2;
+	}
+
+	if (ip_state[0] & BAD_CPU_CLUSTER1) {
+		/* fail entire cluster when one or more core is bad */
+		ip_state[0] |= BAD_CPU_CLUSTER1;
+		fdt_path_del_node(blob, "/cpus/cpu-map/cluster1");
+	}
+
+	if (ip_state[0] & BAD_CPU_CLUSTER2) {
+		/* fail entire cluster when one or more core is bad */
+		ip_state[0] |= BAD_CPU_CLUSTER2;
+		fdt_path_del_node(blob, "/cpus/cpu-map/cluster2");
+	} else if (ip_state[0] & BAD_CPU_CLUSTER1) {
+		/* cluster nodes must be named in a continuous series */
+		fdt_path_set_name(blob, "/cpus/cpu-map/cluster2", "cluster1");
+	}
+
+	/* gpu: ip_state[1]: bit1~4 */
+	if (ip_state[1] & BAD_GPU) {
+		log_debug("fail gpu\n");
+		fdt_status_fail_by_pathf(blob, "/gpu@fb000000");
+	}
+
+	/* rkvdec: ip_state[1]: bit6,7 */
+	if (ip_state[1] & BIT(6)) {
+		fdt_status_fail_by_pathf(blob, "/video-codec@fdc38000");
+		fdt_status_fail_by_pathf(blob, "/video-decoder@fdc38100");
+	}
+	if (ip_state[1] & BIT(7)) {
+		fdt_status_fail_by_pathf(blob, "/video-codec@fdc40000");
+		fdt_status_fail_by_pathf(blob, "/video-decoder@fdc40100");
+	}
+
+	parent = fdt_path_offset(blob, "/cpus");
+	if (parent < 0)
+		return 0;
+
+	/* cpu: ip_state[0]: bit0~7 */
+	for (i = 0; i < 8; i++) {
+		/* fail any bad cpu core */
+		if (!(ip_state[0] & BIT(i)))
+			continue;
+
+		node = fdt_subnode_offset(blob, parent, cpu_node_names[i]);
+		if (node >= 0) {
+			log_debug("fail cpu %s\n", cpu_node_names[i]);
+			fdt_status_fail(blob, node);
+		}
+	}
+
+	return 0;
+}
 #endif
