--- a/src/core/libraries/libc_internal/libc_internal.cpp	2024-12-29 10:12:20.545858744 +0100
+++ b/src/core/libraries/libc_internal/libc_internal.cpp	2024-12-29 10:14:27.615597705 +0100
@@ -88,7 +88,7 @@
 }
 
 float PS4_SYSV_ABI internal_sinf(float x) {
-    return std::sinf(x);
+    return sinf(x);
 }
 
 double PS4_SYSV_ABI internal_cos(double x) {
@@ -96,7 +96,7 @@
 }
 
 float PS4_SYSV_ABI internal_cosf(float x) {
-    return std::cosf(x);
+    return cosf(x);
 }
 
 void PS4_SYSV_ABI internal_sincos(double x, double* sinp, double* cosp) {
@@ -105,8 +105,8 @@
 }
 
 void PS4_SYSV_ABI internal_sincosf(float x, float* sinp, float* cosp) {
-    *sinp = std::sinf(x);
-    *cosp = std::cosf(x);
+    *sinp = sinf(x);
+    *cosp = cosf(x);
 }
 
 double PS4_SYSV_ABI internal_tan(double x) {
@@ -114,7 +114,7 @@
 }
 
 float PS4_SYSV_ABI internal_tanf(float x) {
-    return std::tanf(x);
+    return tanf(x);
 }
 
 double PS4_SYSV_ABI internal_asin(double x) {
@@ -122,7 +122,7 @@
 }
 
 float PS4_SYSV_ABI internal_asinf(float x) {
-    return std::asinf(x);
+    return asinf(x);
 }
 
 double PS4_SYSV_ABI internal_acos(double x) {
@@ -130,7 +130,7 @@
 }
 
 float PS4_SYSV_ABI internal_acosf(float x) {
-    return std::acosf(x);
+    return acosf(x);
 }
 
 double PS4_SYSV_ABI internal_atan(double x) {
@@ -138,7 +138,7 @@
 }
 
 float PS4_SYSV_ABI internal_atanf(float x) {
-    return std::atanf(x);
+    return atanf(x);
 }
 
 double PS4_SYSV_ABI internal_atan2(double y, double x) {
@@ -146,7 +146,7 @@
 }
 
 float PS4_SYSV_ABI internal_atan2f(float y, float x) {
-    return std::atan2f(y, x);
+    return atan2f(y, x);
 }
 
 double PS4_SYSV_ABI internal_exp(double x) {
@@ -154,7 +154,7 @@
 }
 
 float PS4_SYSV_ABI internal_expf(float x) {
-    return std::expf(x);
+    return expf(x);
 }
 
 double PS4_SYSV_ABI internal_exp2(double x) {
@@ -162,7 +162,7 @@
 }
 
 float PS4_SYSV_ABI internal_exp2f(float x) {
-    return std::exp2f(x);
+    return exp2f(x);
 }
 
 double PS4_SYSV_ABI internal_pow(double x, double y) {
@@ -170,7 +170,7 @@
 }
 
 float PS4_SYSV_ABI internal_powf(float x, float y) {
-    return std::powf(x, y);
+    return powf(x, y);
 }
 
 double PS4_SYSV_ABI internal_log(double x) {
@@ -178,7 +178,7 @@
 }
 
 float PS4_SYSV_ABI internal_logf(float x) {
-    return std::logf(x);
+    return logf(x);
 }
 
 double PS4_SYSV_ABI internal_log10(double x) {
@@ -186,7 +186,7 @@
 }
 
 float PS4_SYSV_ABI internal_log10f(float x) {
-    return std::log10f(x);
+    return log10f(x);
 }
 
 void* PS4_SYSV_ABI internal_malloc(size_t size) {
