--- a/core/deps/libzip/mkstemp.c
+++ b/core/deps/libzip/mkstemp.c
@@ -43,9 +43,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#ifdef __APPLE__
 #include <unistd.h>
-#endif
 
 #ifndef O_BINARY
 #define O_BINARY 0
--- a/core/deps/libzip/zip_close.c
+++ b/core/deps/libzip/zip_close.c
@@ -42,9 +42,7 @@
 
 #include "zipint.h"
 
-#ifdef __APPLE__
 #include <unistd.h>
-#endif
 
 static int add_data(struct zip *, struct zip_source *, struct zip_dirent *,
 		    FILE *);
