--- a/projects/unix/Makefile	2025-01-22 19:53:56.441226176 +0100
+++ b/projects/unix/Makefile	2025-01-22 19:54:39.117243843 +0100
@@ -54,6 +54,9 @@
   $(error OS type "$(UNAME)" not supported.  Please file bug report at 'https://github.com/mupen64plus/mupen64plus-core/issues')
 endif
 
+# GCC 14 workaround
+CFLAGS +=-Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types -Wno-error=int-conversion
+
 # detect system architecture
 HOST_CPU ?= $(shell uname -m)
 CPU := NONE
