--- a/src/xenia/kernel/xam/xam_ui.cc	2025-01-21 20:17:07.191762150 +0100
+++ b/src/xenia/kernel/xam/xam_ui.cc	2025-01-21 20:19:27.595973624 +0100
@@ -655,9 +655,7 @@
 
     if (achievement_entry.IsUnlocked()) {
       if (achievement_entry.IsUnlockedOnline()) {
-        ImGui::TextUnformatted(fmt::format("Unlocked: {:%Y-%m-%d %H:%M}",
-                                           achievement_entry.unlock_time)
-                                   .c_str());
+        ImGui::TextUnformatted(fmt::format("Unlocked online").c_str());
       } else {
         ImGui::TextUnformatted(fmt::format("Unlocked: Locally").c_str());
       }
