diff --git a/package/Config.in b/package/Config.in
index 74ca42066c..31099fbefb 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -330,6 +330,7 @@ comment "Graphic applications"
 	source "package/pngquant/Config.in"
 	source "package/qt5cinex/Config.in"
 	source "package/rrdtool/Config.in"
+	source "package/spirv-llvm-translator/Config.in" # reglinux - add spirv-llvm-translator
 	source "package/spirv-tools/Config.in"
 	source "package/stellarium/Config.in"
 	source "package/sway/Config.in"
