diff --git a/package/Config.in.host b/package/Config.in.host
index bf04f5fa974..7d012b03296 100644
--- a/package/Config.in.host
+++ b/package/Config.in.host
@@ -130,4 +130,10 @@ menu "Host utilities"
 	source "package/zip/Config.in.host"
 	source "package/zstd/Config.in.host"
 
+	# reglinux
+	source "package/llvm-project/llvm-cmake/Config.in.host"
+	source "package/python-pyelftools/Config.in.host"
+	source "package/python-pylibfdt/Config.in.host"
+	source "package/python-setuptools-scm/Config.in.host"
+
 endmenu
