diff --git a/package/busybox/busybox.mk b/package/busybox/busybox.mk
index d7eda71db6..a3ad46da85 100644
--- a/package/busybox/busybox.mk
+++ b/package/busybox/busybox.mk
@@ -437,6 +437,11 @@ define BUSYBOX_KCONFIG_FIXUP_CMDS
 	$(PACKAGES_BUSYBOX_CONFIG_FIXUPS)
 endef
 
+# reglinux
+define BUSYBOX_INSTALL_SEEDRNG
+	cd $(TARGET_DIR)/bin && ln -sf /bin/busybox seedrng
+endef
+
 define BUSYBOX_BUILD_CMDS
 	$(BUSYBOX_MAKE_ENV) $(MAKE) $(BUSYBOX_MAKE_OPTS) -C $(@D)
 endef
@@ -450,6 +455,7 @@ define BUSYBOX_INSTALL_TARGET_CMDS
 	$(BUSYBOX_INSTALL_UDHCPC_SCRIPT)
 	$(BUSYBOX_INSTALL_ZCIP_SCRIPT)
 	$(BUSYBOX_INSTALL_MDEV_CONF)
+	$(BUSYBOX_INSTALL_SEEDRNG)
 endef
 
 # Install the sysvinit scripts, for the moment, but not those that already
