diff --git a/package/busybox/busybox.mk b/package/busybox/busybox.mk
index d61fed9879..01c0d1c5c8 100644
--- a/package/busybox/busybox.mk
+++ b/package/busybox/busybox.mk
@@ -424,6 +424,11 @@ define BUSYBOX_KCONFIG_FIXUP_CMDS
 	$(BUSYBOX_SET_INDIVIDUAL_BINARIES)
 endef
 
+# reglinux
+define BUSYBOX_INSTALL_SEEDRNG
+	cd $(TARGET_DIR)/bin && ln -sf /bin/busybox seedrng
+endef
+
 define BUSYBOX_BUILD_CMDS
 	$(BUSYBOX_MAKE_ENV) $(MAKE) $(BUSYBOX_MAKE_OPTS) -C $(@D)
 endef
@@ -437,6 +442,7 @@ define BUSYBOX_INSTALL_TARGET_CMDS
 	$(BUSYBOX_INSTALL_UDHCPC_SCRIPT)
 	$(BUSYBOX_INSTALL_ZCIP_SCRIPT)
 	$(BUSYBOX_INSTALL_MDEV_CONF)
+	$(BUSYBOX_INSTALL_SEEDRNG)
 endef
 
 # Install the sysvinit scripts, for the moment, but not those that already
