--- a/src/expansion.cpp	2025-05-06 11:15:41.359878682 +0200
+++ b/src/expansion.cpp	2025-05-06 12:13:40.342423902 +0200
@@ -2760,8 +2760,7 @@
 		NULL, cdtvsram_init, NULL, NULL, ROMTYPE_CDTVSRAM | ROMTYPE_NOT, 0, 0, BOARD_NONAUTOCONFIG_BEFORE, true,
 		NULL, 0,
 		false, EXPANSIONTYPE_INTERNAL,
-		0, 0, 0, false, NULL,
-		false, 0, cdtvsram_settings
+		cdtvsram_settings
 	},
 	//{
 	//	_T("cdtvcr"), _T("CDTV-CR"), _T("Commodore"),
@@ -2786,7 +2785,6 @@
 		NULL, gayle_ide_init, NULL, gayle_add_ide_unit, ROMTYPE_MB_IDE | ROMTYPE_NOT, 0, 0, BOARD_NONAUTOCONFIG_BEFORE, true,
 		NULL, 0,
 		false, EXPANSIONTYPE_INTERNAL | EXPANSIONTYPE_IDE,
-		0, 0, 0, false, NULL, false, 1
 	},
 	{
 		_T("pcmcia_mb"), _T("A600/A1200 PCMCIA"), _T("Commodore"),
@@ -2794,7 +2792,6 @@
 		NULL, 0,
 		false, EXPANSIONTYPE_INTERNAL
 	},
-
 	{
 		NULL
 	}
