diff --git a/package/opentyrian-data/Config.in b/package/opentyrian-data/Config.in
index c0698c297c8..a1a1e1a4a90 100644
--- a/package/opentyrian-data/Config.in
+++ b/package/opentyrian-data/Config.in
@@ -1,6 +1,7 @@
 config BR2_PACKAGE_OPENTYRIAN_DATA
 	bool "OpenTyrian data"
-	depends on BR2_PACKAGE_OPENTYRIAN
+	# batocera
+	#depends on BR2_PACKAGE_OPENTYRIAN
 	help
 	  Data files to use with OpenTyrian game. Currently only the
 	  original Tyrian 2.1 data files, that have been released as
