diff --git a/package/llvm-project/llvm-cmake/llvm-cmake.mk b/package/llvm-project/llvm-cmake/llvm-cmake.mk
index 33b6ba8339b..1be63a9cbe7 100644
--- a/package/llvm-project/llvm-cmake/llvm-cmake.mk
+++ b/package/llvm-project/llvm-cmake/llvm-cmake.mk
@@ -10,8 +10,9 @@ LLVM_CMAKE_SOURCE = cmake-$(LLVM_CMAKE_VERSION).src.tar.xz
 LLVM_CMAKE_LICENSE = Apache-2.0 with exceptions
 
 define HOST_LLVM_CMAKE_INSTALL_CMDS
-	mkdir -p $(HOST_DIR)/lib/cmake/llvm
-	cp -Rv $(@D)/Modules/* $(HOST_DIR)/lib/cmake/llvm
+    # batocera - fix Modules directory below...
+	mkdir -p $(HOST_DIR)/lib/cmake/llvm/Modules
+	cp -Rv $(@D)/Modules/* $(HOST_DIR)/lib/cmake/llvm/Modules
 endef
 
 $(eval $(host-generic-package))
