diff --git a/package/llvm-project/llvm-cmake/llvm-cmake.hash b/package/llvm-project/llvm-cmake/llvm-cmake.hash
index 756404cdd9..14ca9a26a9 100644
--- a/package/llvm-project/llvm-cmake/llvm-cmake.hash
+++ b/package/llvm-project/llvm-cmake/llvm-cmake.hash
@@ -1,2 +1,3 @@
 # locally calculated
 sha256  8986f29b634fdaa9862eedda78513969fe9788301c9f2d938f4c10a3e7a3e7ea  cmake-15.0.7.src.tar.xz
+sha256  11c5a28f90053b0c43d0dec3d0ad579347fc277199c005206b963c19aae514e3  cmake-19.1.7.src.tar.xz
