diff --git a/package/llvm-project/llvm-cmake/llvm-cmake.hash b/package/llvm-project/llvm-cmake/llvm-cmake.hash
index 862c5d92e8..dc35ab306f 100644
--- a/package/llvm-project/llvm-cmake/llvm-cmake.hash
+++ b/package/llvm-project/llvm-cmake/llvm-cmake.hash
@@ -1,2 +1,3 @@
 # locally calculated
 sha256  21cf3f52c53dc8b8972122ae35a5c18de09c7df693b48b5cd8553c3e3fed090d  cmake-15.0.3.src.tar.xz
+sha256  11c5a28f90053b0c43d0dec3d0ad579347fc277199c005206b963c19aae514e3  cmake-19.1.7.src.tar.xz
