--- a/z80.cpu	2025-02-04 07:28:20.000000000 +0100
+++ b/z80.cpu	2025-02-07 15:28:21.393132891 +0100
@@ -1544,7 +1544,7 @@
 	or tmp last_flag_result last_flag_result
 	update_flags H0N0C1
 
-00000000 nop
+00000000 z80_nop
 
 01110110 halt
 	cmp nmi_cycle cycles
--- a/m68k.cpu	2025-02-04 07:28:20.000000000 +0100
+++ b/m68k.cpu	2025-02-07 15:28:39.047499874 +0100
@@ -2134,7 +2134,7 @@
 	cycles 128
 	m68k_prefetch
 
-0100111001110001 nop
+0100111001110001 m68k_nop
 	m68k_prefetch
 
 0100111001110011 rte
