diff --git a/package/cmake/cmake.mk b/package/cmake/cmake.mk
index 0c5d46c23d..fd8e90ab0b 100644
--- a/package/cmake/cmake.mk
+++ b/package/cmake/cmake.mk
@@ -61,7 +61,7 @@ define HOST_CMAKE_CONFIGURE_CMDS
 			-DCMAKE_C_FLAGS="$(HOST_CMAKE_CFLAGS)" \
 			-DCMAKE_CXX_FLAGS="$(HOST_CMAKE_CXXFLAGS)" \
 			-DCMAKE_EXE_LINKER_FLAGS="$(HOST_LDFLAGS)" \
-			-DCMAKE_USE_OPENSSL:BOOL=OFF \
+			-DCMAKE_USE_OPENSSL:BOOL=ON \
 			-DBUILD_CursesDialog=OFF \
 	)
 endef
