diff --git a/package/cmake/cmake.mk b/package/cmake/cmake.mk
index 1eb4da90ad..df5bd57f8f 100644
--- a/package/cmake/cmake.mk
+++ b/package/cmake/cmake.mk
@@ -62,7 +62,7 @@ define HOST_CMAKE_CONFIGURE_CMDS
 			-DCMAKE_CXX_FLAGS="$(HOST_CMAKE_CXXFLAGS)" \
 			-DCMAKE_EXE_LINKER_FLAGS="$(HOST_LDFLAGS)" \
 			-DCMAKE_USE_OPENSSL:BOOL=OFF \
-			-DBUILD_CursesDialog=OFF \
+			-DBUILD_CursesDialog=ON \
 	)
 endef
 
