--- a/CMakeLists.txt	2026-01-04 16:21:04.546600908 +0100
+++ b/CMakeLists.txt	2026-01-04 16:21:20.818418907 +0100
@@ -248,12 +248,7 @@
     #add_definitions(-pipe -march=native)
 endif()
 if(ARM_DYNAREC)
-    if (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS "9.0")
-        set(ASMFLAGS  -pipe -march=armv8.2-a+crc+simd+crypto)
-    else()
-        #putting a random cpu that have 8.2 architecture so assembly can be build. the use of the feature are conditionnal anyway
-        set(ASMFLAGS  -pipe -mcpu=cortex-a76)
-    endif()
+    set(ASMFLAGS  -pipe -march=armv8.2-a+crc+simd+crypto)
 endif()
 if(ANDROID)
     add_definitions(-DANDROID)
