diff --git a/package/wlroots/Config.in b/package/wlroots/Config.in
index d81506c375..b6299d8dac 100644
--- a/package/wlroots/Config.in
+++ b/package/wlroots/Config.in
@@ -22,6 +22,10 @@ config BR2_PACKAGE_WLROOTS
 	select BR2_PACKAGE_LIBXKBCOMMON
 	select BR2_PACKAGE_PIXMAN
 	select BR2_PACKAGE_SEATD
+    select BR2_PACKAGE_HWDATA
+    select BR2_PACKAGE_HWDATA_PNP_IDS
+    select BR2_PACKAGE_LIBDISPLAY_INFO
+    select BR2_PACKAGE_LIBLIFTOFF
 	select BR2_PACKAGE_WAYLAND
 	select BR2_PACKAGE_WAYLAND_PROTOCOLS
 	help
