diff --git a/package/wlroots/wlroots.mk b/package/wlroots/wlroots.mk
index e0f1e1053e..d2f89707ce 100644
--- a/package/wlroots/wlroots.mk
+++ b/package/wlroots/wlroots.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-WLROOTS_VERSION = 0.18.2
+WLROOTS_VERSION = 0.19.2
 WLROOTS_SITE = https://gitlab.freedesktop.org/wlroots/wlroots/-/releases/$(WLROOTS_VERSION)/downloads
 WLROOTS_LICENSE = MIT
 WLROOTS_LICENSE_FILES = LICENSE
@@ -31,6 +31,10 @@ WLROOTS_CONF_OPTS = -Dexamples=false -Dxcb-errors=disabled
 WLROOTS_RENDERERS = gles2
 WLROOTS_BACKENDS = libinput drm
 
+ifeq ($(BR2_PACKAGE_HAS_LIBGBM),y)
+WLROOTS_CONF_OPTS += -Dallocators=gbm
+endif
+
 ifeq ($(BR2_PACKAGE_WLROOTS_X11),y)
 WLROOTS_BACKENDS += x11
 WLROOTS_DEPENDENCIES += libxcb xcb-util-wm xcb-util-renderutil xlib_libX11
@@ -43,7 +47,12 @@ else
 WLROOTS_CONF_OPTS += -Dxwayland=disabled
 endif
 
-ifeq ($(BR2_PACKAGE_MESA3D_VULKAN_DRIVER)$(BR2_PACKAGE_VULKAN_LOADER),yy)
+# reglinux - add vulkan build dependency
+ifeq ($(BR2_PACKAGE_VULKAN_HEADERS)$(BR2_PACKAGE_VULKAN_LOADER),yy)
+    WLROOTS_DEPENDENCIES +=  vulkan-headers vulkan-loader host-glslang
+endif
+
+ifeq ($(BR2_PACKAGE_MESA3D_VULKAN_DRIVER),y)
 WLROOTS_RENDERERS += vulkan
 WLROOTS_DEPENDENCIES += mesa3d vulkan-loader
 endif
