diff --git a/package/dropbear/S50dropbear b/package/dropbear/S50dropbear
index 9474eaafaa..335ea278f9 100644
--- a/package/dropbear/S50dropbear
+++ b/package/dropbear/S50dropbear
@@ -3,10 +3,17 @@
 # Starts dropbear sshd.
 #
 
+# REG-Linux
+if [ "$(/usr/bin/system-settings-get system.ssh.enabled)" = "0" ]; then
+    echo "SSH services are disabled"
+    exit
+fi
+
 # Allow a few customizations from a config file
 test -r /etc/default/dropbear && . /etc/default/dropbear
 
 start() {
+
 	DROPBEAR_ARGS="$DROPBEAR_ARGS -R"
 
 	# If /etc/dropbear is a symlink to /var/run/dropbear, and
@@ -33,10 +40,11 @@ start() {
 		--exec /usr/sbin/dropbear -- $DROPBEAR_ARGS
 	[ $? = 0 ] && echo "OK" || echo "FAIL"
 }
+
 stop() {
 	printf "Stopping dropbear sshd: "
 	start-stop-daemon -K -q -p /var/run/dropbear.pid
-	[ $? = 0 ] && echo "OK" || echo "FAIL"
+	pkill -f /usr/sbin/dropbear
 }
 restart() {
 	stop
