diff --git a/package/x11r7/libxcb/libxcb.mk b/package/x11r7/libxcb/libxcb.mk
index 4b293f9801c..458996e8f9b 100644
--- a/package/x11r7/libxcb/libxcb.mk
+++ b/package/x11r7/libxcb/libxcb.mk
@@ -3,10 +3,10 @@
 # libxcb
 #
 ################################################################################
-
-LIBXCB_VERSION = 1.15
+# reglinux - update
+LIBXCB_VERSION = 1.16
 LIBXCB_SOURCE = libxcb-$(LIBXCB_VERSION).tar.xz
-LIBXCB_SITE = http://xcb.freedesktop.org/dist
+LIBXCB_SITE = https://xorg.freedesktop.org/archive/individual/lib
 LIBXCB_LICENSE = MIT
 LIBXCB_LICENSE_FILES = COPYING
 LIBXCB_CPE_ID_VENDOR = x
