diff --git a/package/qt6/qt6tools/qt6tools.mk b/package/qt6/qt6tools/qt6tools.mk
index b063f08887..cb8332d6ff 100644
--- a/package/qt6/qt6tools/qt6tools.mk
+++ b/package/qt6/qt6tools/qt6tools.mk
@@ -27,11 +27,15 @@ QT6TOOLS_LICENSE_FILES = \
 	LICENSES/LGPL-3.0-only.txt \
 	LICENSES/Qt-GPL-exception-1.0.txt
 
+# REGLINUX adjust options
 QT6TOOLS_CONF_OPTS = \
 	-DQT_HOST_PATH=$(HOST_DIR) \
 	-DBUILD_WITH_PCH=OFF \
 	-DQT_BUILD_EXAMPLES=OFF \
-	-DQT_BUILD_TESTS=OFF
+	-DQT_BUILD_TESTS=OFF \
+	-DQT_BUILD_TESTS_BY_DEFAULT=OFF \
+	-DQT_BUILD_EXAMPLES_BY_DEFAULT=OFF \
+	-DQT_INSTALL_EXAMPLES_SOURCES_BY_DEFAULT=OFF
 
 QT6TOOLS_DEPENDENCIES = \
 	qt6base \
@@ -41,9 +45,17 @@ ifeq ($(BR2_PACKAGE_QT6DECLARATIVE),y)
 QT6TOOLS_DEPENDENCIES += qt6declarative
 endif
 
+# REGLINUX adjust options for host-qt6tools
 HOST_QT6TOOLS_CONF_OPTS = \
 	-DQT_BUILD_EXAMPLES=OFF \
-	-DQT_BUILD_TESTS=OFF
+	-DQT_BUILD_TESTS=OFF \
+	-DFEATURE_qdoc=OFF \
+	-DFEATURE_clang=OFF \
+	-DQT_FEATURE_png=ON \
+	-DFEATURE_pixeltool=ON \
+	-DFEATURE_designer=ON \
+	-DQT_BUILD_EXAMPLES_BY_DEFAULT=OFF \
+	-DQT_BUILD_TESTS_BY_DEFAULT=OFF
 
 HOST_QT6TOOLS_DEPENDENCIES = host-qt6base
 
