--- a/externals/nx_tzdb/CMakeLists.txt	2024-06-05 02:48:29.084565884 +0200
+++ b/externals/nx_tzdb/CMakeLists.txt	2024-06-05 02:48:45.616911799 +0200
@@ -20,12 +20,12 @@
 if (NOT DATE_PROG)
     set(CAN_BUILD_NX_TZDB false)
 endif()
-if (CMAKE_SYSTEM_NAME STREQUAL "Windows" OR ANDROID)
+#if (CMAKE_SYSTEM_NAME STREQUAL "Windows" OR ANDROID)
     # tzdb_to_nx currently requires a posix-compliant host
     # MinGW and Android are handled here due to the executable format being different from the host system
     # TODO (lat9nq): cross-compiling support
     set(CAN_BUILD_NX_TZDB false)
-endif()
+#endif()
 
 set(NX_TZDB_VERSION "221202")
 set(NX_TZDB_ARCHIVE "${CMAKE_CURRENT_BINARY_DIR}/${NX_TZDB_VERSION}.zip")
