diff --git a/deps/switchres/custom_video_drmkms.cpp b/deps/switchres/custom_video_drmkms.cpp
index f71c49c..9301948 100755
--- a/deps/switchres/custom_video_drmkms.cpp
+++ b/deps/switchres/custom_video_drmkms.cpp
@@ -532,7 +532,21 @@ bool drmkms_timing::init()
 	char drm_name[15] = "/dev/dri/card_";
 	drmModeRes *p_res;
 	drmModeConnector *p_connector;
-
+	int drmConn = 0;
+
+   // batocera
+   {
+     FILE* fdDrmConn;
+     int drmConnRead;
+     if((fdDrmConn = fopen("/var/run/drmConn", "r")) != NULL) {
+       if(fscanf(fdDrmConn, "%i", &drmConnRead) == 1) {
+         if(drmConnRead>=0 && drmConn<p_res->count_connectors) {
+            drmConn = drmConnRead;
+         }
+       }
+     }
+   }
+   //
 	int output_position = 0;
 	for (int num = 0; !m_desktop_output && num < MAX_CARD_ID; num++)
 	{
diff --git a/gfx/common/drm_common.c b/gfx/common/drm_common.c
index 5759d82..e18214a 100644
--- a/gfx/common/drm_common.c
+++ b/gfx/common/drm_common.c
@@ -68,6 +68,21 @@ bool drm_get_connector(int fd, unsigned monitor_index)
    unsigned i;
    unsigned monitor_index_count = 0;
    unsigned monitor       = MAX(monitor_index, 1);
+   int drmConn = 0;
+
+   // batocera
+   {
+     FILE* fdDrmConn;
+     int drmConnRead;
+     if((fdDrmConn = fopen("/var/run/drmConn", "r")) != NULL) {
+       if(fscanf(fdDrmConn, "%i", &drmConnRead) == 1) {
+    if(drmConnRead>=0 && drmConn<g_drm_resources->count_connectors) {
+      drmConn = drmConnRead;
+    }
+       }
+     }
+   }
+   //
 
    /* Enumerate all connectors. */
 
@@ -75,7 +90,11 @@ bool drm_get_connector(int fd, unsigned monitor_index)
 
    for (i = 0; (int)i < g_drm_resources->count_connectors; i++)
    {
-      drmModeConnectorPtr conn = drmModeGetConnector(
+      drmModeConnectorPtr conn;
+
+      if(i != drmConn) continue;
+
+      conn = drmModeGetConnector(
             fd, g_drm_resources->connectors[i]);
 
       if (conn)
@@ -96,6 +115,8 @@ bool drm_get_connector(int fd, unsigned monitor_index)
 
    for (i = 0; (int)i < g_drm_resources->count_connectors; i++)
    {
+      if(i != drmConn) continue;
+
       g_drm_connector = drmModeGetConnector(fd,
             g_drm_resources->connectors[i]);
 
diff --git a/gfx/drivers/drm_gfx.c b/gfx/drivers/drm_gfx.c
index ac8d794..21c397a 100644
--- a/gfx/drivers/drm_gfx.c
+++ b/gfx/drivers/drm_gfx.c
@@ -600,6 +600,7 @@ static bool init_drm(void)
 {
    uint i;
    drmModeConnector *connector;
+   int drmConn = 0;
 
    drm.fd = open("/dev/dri/card0", O_RDWR);
 
@@ -633,9 +634,25 @@ static bool init_drm(void)
       return false;
    }
 
+   // batocera
+   {
+     FILE* fdDrmConn;
+     int drmConnRead;
+     if((fdDrmConn = fopen("/var/run/drmConn", "r")) != NULL) {
+       if(fscanf(fdDrmConn, "%i", &drmConnRead) == 1) {
+    if(drmConnRead>=0 && drmConn<(uint)drm_resources->count_connectors) {
+      drmConn = drmConnRead;
+    }
+       }
+     }
+   }
+   //
+
    /* Find a connected connector. */
    for (i = 0; i < (uint)drm.resources->count_connectors; i++)
    {
+      if(i != drmConn) continue;
+
       connector = drmModeGetConnector(drm.fd, drm.resources->connectors[i]);
       /* It's connected, let's use it. */
       if (connector->connection == DRM_MODE_CONNECTED)
