diff --git a/package/x11r7/xlib_libXrandr/xlib_libXrandr.mk b/package/x11r7/xlib_libXrandr/xlib_libXrandr.mk
index 5e88173ed7b..6d3dcf3f88c 100644
--- a/package/x11r7/xlib_libXrandr/xlib_libXrandr.mk
+++ b/package/x11r7/xlib_libXrandr/xlib_libXrandr.mk
@@ -3,8 +3,8 @@
 # xlib_libXrandr
 #
 ################################################################################
-
-XLIB_LIBXRANDR_VERSION = 1.5.3
+# reglinux - update
+XLIB_LIBXRANDR_VERSION = 1.5.4
 XLIB_LIBXRANDR_SOURCE = libXrandr-$(XLIB_LIBXRANDR_VERSION).tar.xz
 XLIB_LIBXRANDR_SITE = https://xorg.freedesktop.org/archive/individual/lib
 XLIB_LIBXRANDR_LICENSE = MIT
