diff --git a/package/sway/Config.in b/package/sway/Config.in
index d78f792c488..968375583df 100644
--- a/package/sway/Config.in
+++ b/package/sway/Config.in
@@ -1,6 +1,5 @@
 config BR2_PACKAGE_SWAY
 	bool "sway"
-	depends on BR2_PACKAGE_SYSTEMD # is required by the sd-bus provider
 	depends on !BR2_STATIC_LIBS # wlroots
 	depends on BR2_TOOLCHAIN_HAS_THREADS # pango, wlroots
 	depends on BR2_PACKAGE_HAS_LIBEGL # wlroots
@@ -18,6 +17,7 @@ config BR2_PACKAGE_SWAY
 	select BR2_PACKAGE_CAIRO_PNG
 	select BR2_PACKAGE_PANGO
 	select BR2_PACKAGE_XKEYBOARD_CONFIG # runtime
+	select BR2_PACKAGE_LIBDECOR
 	help
 	  i3-compatible Wayland compositor
 	  Note:
