diff --git a/package/sway/sway.mk b/package/sway/sway.mk
index 74614b0c7f4..09c45769c0a 100644
--- a/package/sway/sway.mk
+++ b/package/sway/sway.mk
@@ -4,17 +4,16 @@
 #
 ################################################################################
 
-SWAY_VERSION = 1.10
+SWAY_VERSION = 1.11
 SWAY_SITE = https://github.com/swaywm/sway/releases/download/$(SWAY_VERSION)
 SWAY_LICENSE = MIT
 SWAY_LICENSE_FILES = LICENSE
-SWAY_DEPENDENCIES = systemd host-pkgconf wlroots json-c pcre cairo pango
+SWAY_DEPENDENCIES = host-pkgconf wlroots json-c pcre cairo pango
 SWAY_CONF_OPTS = \
 	-Dwerror=false \
 	-Dzsh-completions=false \
 	-Dfish-completions=false \
-	-Dman-pages=disabled \
-	-Dsd-bus-provider=libsystemd
+	-Dman-pages=disabled
 
 ifeq ($(BR2_PACKAGE_GDK_PIXBUF),y)
 SWAY_CONF_OPTS += -Dgdk-pixbuf=enabled
@@ -53,4 +52,11 @@ else
 SWAY_CONF_OPTS += -Dtray=disabled
 endif
 
+# Install only what is needed avoiding systemd files
+define SWAY_INSTALL_TARGET_CMDS
+	mkdir -p $(TARGET_DIR)/usr/bin
+	$(INSTALL) -D $(@D)/buildroot-build/sway/sway         $(TARGET_DIR)/usr/bin
+	$(INSTALL) -D $(@D)/buildroot-build/swaymsg/swaymsg   $(TARGET_DIR)/usr/bin
+endef
+
 $(eval $(meson-package))
