diff --git a/package/sway/sway.mk b/package/sway/sway.mk
index 0aad9de712..5eeb0fe3f7 100644
--- a/package/sway/sway.mk
+++ b/package/sway/sway.mk
@@ -3,18 +3,23 @@
 # sway
 #
 ################################################################################
-
-SWAY_VERSION = 1.8.1
+# batocera (update)
+SWAY_VERSION = 1.9
 SWAY_SITE = https://github.com/swaywm/sway/releases/download/$(SWAY_VERSION)
 SWAY_LICENSE = MIT
 SWAY_LICENSE_FILES = LICENSE
-SWAY_DEPENDENCIES = systemd host-pkgconf wlroots json-c pcre cairo pango
+SWAY_DEPENDENCIES = host-pkgconf wlroots json-c pcre2 cairo pango
 SWAY_CONF_OPTS = \
 	-Dwerror=false \
 	-Dzsh-completions=false \
+	-Dbash-completions=true \
 	-Dfish-completions=false \
-	-Dman-pages=disabled \
-	-Dsd-bus-provider=libsystemd
+	-Dswaybar=false \
+	-Dswaynag=false \
+	-Dtray=disabled \
+	-Dman-pages=disabled
+# batocera - We don't want systemd
+#-Dsd-bus-provider=libsystemd
 
 ifeq ($(BR2_PACKAGE_WLROOTS_X11),y)
 SWAY_CONF_OPTS += -Dxwayland=enabled
@@ -59,4 +64,11 @@ else
 SWAY_CONF_OPTS += -Dtray=disabled
 endif
 
+# batocera - Install only what is needed avoiding systemd files
+define SWAY_INSTALL_TARGET_CMDS
+    mkdir -p $(TARGET_DIR)/usr/bin
+    $(INSTALL) -D $(@D)/build/sway/sway         $(TARGET_DIR)/usr/bin
+    $(INSTALL) -D $(@D)/build/swaymsg/swaymsg   $(TARGET_DIR)/usr/bin
+endef
+
 $(eval $(meson-package))
