diff --git a/package/libcurl/libcurl.mk b/package/libcurl/libcurl.mk
index 2066ba0388..250ef55e11 100644
--- a/package/libcurl/libcurl.mk
+++ b/package/libcurl/libcurl.mk
@@ -202,3 +202,17 @@ LIBCURL_POST_INSTALL_TARGET_HOOKS += LIBCURL_TARGET_CLEANUP
 endif
 
 $(eval $(autotools-package))
+
+# reglinux - we need host-libcurl for several packages
+HOST_LIBCURL_DEPENDENCIES = host-openssl
+HOST_LIBCURL_CONF_OPTS = \
+	--disable-manual \
+	--disable-ntlm-wb \
+	--disable-curldebug \
+	--with-ssl \
+	--without-gnutls \
+	--without-mbedtls \
+	--without-libpsl \
+	--without-nss
+
+$(eval $(host-autotools-package))
