diff --git a/package/qemu/qemu.mk b/package/qemu/qemu.mk
index 3a2f69ea3c3..595411b50c5 100644
--- a/package/qemu/qemu.mk
+++ b/package/qemu/qemu.mk
@@ -528,3 +528,11 @@ $(eval $(host-generic-package))
 
 # variable used by other packages
 QEMU_USER = $(HOST_DIR)/bin/qemu-$(HOST_QEMU_ARCH)
+
+# reglinux - fixup for mipsel
+ifeq ($(BR2_mipsel),y)
+define HOST_QEMU_POST_FIX_MIPSEL_SYMLINK
+	ln -sf $(HOST_DIR)/bin/qemu-mipsel $(HOST_DIR)/bin/qemu-mips
+endef
+HOST_QEMU_POST_INSTALL_HOOKS += HOST_QEMU_POST_FIX_MIPSEL_SYMLINK
+endif
