diff --git a/package/binutils/binutils.mk b/package/binutils/binutils.mk
index 2b5064d739..eb853b4e97 100644
--- a/package/binutils/binutils.mk
+++ b/package/binutils/binutils.mk
@@ -149,5 +149,17 @@ define HOST_BINUTILS_FIXUP_HARDLINKS
 endef
 HOST_BINUTILS_POST_INSTALL_HOOKS += HOST_BINUTILS_FIXUP_HARDLINKS
 
+# reglinux
+BINUTILSUSR_TOOLS = strings
+define HOST_BINUTILSUSR_FIXUP_HARDLINKS
+	$(foreach tool,$(BINUTILSUSR_TOOLS),\
+		rm -f $(HOST_DIR)/$(GNU_TARGET_NAME)/usr/bin/$(tool) && \
+		mkdir -p $(HOST_DIR)/$(GNU_TARGET_NAME)/sysroot/usr/bin/ && \
+		cp -a $(HOST_DIR)/bin/$(GNU_TARGET_NAME)-$(tool) \
+			$(HOST_DIR)/$(GNU_TARGET_NAME)/sysroot/usr/bin/$(tool)
+	)
+endef
+HOST_BINUTILS_POST_INSTALL_HOOKS += HOST_BINUTILSUSR_FIXUP_HARDLINKS
+
 $(eval $(autotools-package))
 $(eval $(host-autotools-package))
