diff --git a/package/espeak/espeak.mk b/package/espeak/espeak.mk
index 44893ac9be3..fd4c506dc35 100644
--- a/package/espeak/espeak.mk
+++ b/package/espeak/espeak.mk
@@ -11,6 +11,9 @@ ESPEAK_SITE = http://downloads.sourceforge.net/project/espeak/espeak/espeak-$(ES
 ESPEAK_LICENSE = GPL-3.0+
 ESPEAK_LICENSE_FILES = License.txt
 
+# reglinux
+ESPEAK_INSTALL_STAGING = YES
+
 ifeq ($(BR2_PACKAGE_ESPEAK_AUDIO_BACKEND_ALSA),y)
 ESPEAK_AUDIO_BACKEND = portaudio
 ESPEAK_DEPENDENCIES += portaudio
@@ -41,4 +44,9 @@ define ESPEAK_INSTALL_TARGET_CMDS
 		AUDIO="$(ESPEAK_AUDIO_BACKEND)" DESTDIR="$(TARGET_DIR)" install
 endef
 
+# reglinux
+define ESPEAK_INSTALL_STAGING_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/src DESTDIR="$(STAGING_DIR)" install
+endef
+
 $(eval $(generic-package))
