--- a/librmx/source/rmxmedia_externals.h	2024-01-30 04:47:54.000000000 +0100
+++ b/librmx/source/rmxmedia_externals.h	2024-04-17 15:01:37.308640875 +0200
@@ -46,10 +46,17 @@
 
 #elif defined(PLATFORM_LINUX)
 	#if defined(RMX_LINUX_ENFORCE_GLES2)	// Build option: Use OpenGL ES 2
-		#define ALLOW_LEGACY_OPENGL
-		#define RMX_USE_GLES2
-		#include <GLES3/gl3.h>		// We need the ES 3 headers for e.g. glBindVertexArray
-		#include <GLES3/gl3ext.h>
+		//#define ALLOW_LEGACY_OPENGL
+ 		#define RMX_USE_GLES2
+		#include <GLES3/gl3platform.h>
+		#include <GLES3/gl3.h>
+		//#define GL_GLEXT_PROTOTYPES
+		//#include <GLES3/gl3ext.h>
+		#define GL_RGB8		     GL_RGB
+		#define GL_RGBA8	     GL_RGBA
+		//#define glGenVertexArrays		glGenVertexArraysOES
+		//#define glDeleteVertexArrays	glDeleteVertexArraysOES
+		//#define glBindVertexArray		glBindVertexArrayOES
 	#else
 		#define RMX_USE_GLEW
 	#endif
