diff -u -r -N a/RSDKv3/Audio.hpp b/RSDKv3/Audio.hpp
--- a/RSDKv3/Audio.hpp	2025-01-19 17:34:21.803007221 +0100
+++ b/RSDKv3/Audio.hpp	2025-01-19 17:33:05.503102407 +0100
@@ -6,7 +6,7 @@
 #include <vorbis/vorbisfile.h>
 
 #if RETRO_PLATFORM != RETRO_VITA && RETRO_PLATFORM != RETRO_OSX
-#include "SDL.h"
+#include "SDL2/SDL.h"
 #endif
 
 #if RETRO_USING_SDL1 || RETRO_USING_SDL2
diff -u -r -N a/RSDKv3/Drawing.cpp b/RSDKv3/Drawing.cpp
--- a/RSDKv3/Drawing.cpp	2025-01-19 17:34:21.805007194 +0100
+++ b/RSDKv3/Drawing.cpp	2025-01-19 17:33:05.505102377 +0100
@@ -265,7 +265,7 @@
 
     SDL_GL_SetSwapInterval(Engine.vsync ? 1 : 0);
 
-#if RETRO_PLATFORM != RETRO_ANDROID && RETRO_PLATFORM != RETRO_OSX
+#if RETRO_PLATFORM != RETRO_ANDROID && RETRO_PLATFORM != RETRO_OSX && RETRO_PLATFORM != RETRO_LINUX
     // glew Setup
     GLenum err = glewInit();
     if (err != GLEW_OK && err != GLEW_ERROR_NO_GLX_DISPLAY) {
diff -u -r -N a/RSDKv3/RetroEngine.hpp b/RSDKv3/RetroEngine.hpp
--- a/RSDKv3/RetroEngine.hpp	2025-01-19 17:34:21.807007166 +0100
+++ b/RSDKv3/RetroEngine.hpp	2025-01-19 17:33:05.513102257 +0100
@@ -138,13 +138,16 @@
 #endif
 
 #if RETRO_USING_OPENGL
-#if RETRO_PLATFORM == RETRO_ANDROID
+#if RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_LINUX
 #define GL_GLEXT_PROTOTYPES
 
 #include <GLES/gl.h>
 #include <GLES/glext.h>
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
 
-#undef glGenFramebuffers
+#define glOrtho		glOrthof
+/*#undef glGenFramebuffers
 #undef glBindFramebuffers
 #undef glFramebufferTexture2D
 
@@ -160,7 +163,7 @@
 
 #define GL_FRAMEBUFFER         GL_FRAMEBUFFER_OES
 #define GL_COLOR_ATTACHMENT0   GL_COLOR_ATTACHMENT0_OES
-#define GL_FRAMEBUFFER_BINDING GL_FRAMEBUFFER_BINDING_OES
+#define GL_FRAMEBUFFER_BINDING GL_FRAMEBUFFER_BINDING_OES*/
 #elif RETRO_PLATFORM == RETRO_OSX
 #define GL_GLEXT_PROTOTYPES
 #define GL_SILENCE_DEPRECATION
@@ -343,7 +346,7 @@
 
 #if RETRO_PLATFORM == RETRO_WIN || RETRO_PLATFORM == RETRO_UWP || RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_LINUX
 #if RETRO_USING_SDL2
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #elif RETRO_USING_SDL1
 #include <SDL.h>
 #endif
--- a/platforms/Linux.cmake	2025-11-05 11:57:15.958435403 +0100
+++ b/platforms/Linux.cmake	2025-11-05 12:00:57.851284185 +0100
@@ -42,17 +42,7 @@
 endif()
 
 if(RETRO_USE_HW_RENDER)
-    pkg_check_modules(GLEW glew)
-
-    if(NOT GLEW_FOUND)
-        message(NOTICE "could not find glew, attempting to build from source")
-
-    else()
-        message("found GLEW")
-        target_link_libraries(RetroEngine ${GLEW_STATIC_LIBRARIES})
-        target_link_options(RetroEngine PRIVATE ${GLEW_STATIC_LDLIBS_OTHER})
-        target_compile_options(RetroEngine PRIVATE ${GLEW_STATIC_CFLAGS})
-    endif()
+    target_link_libraries(RetroEngine GLESv1_CM GLESv2 EGL)
 endif()
 
 if(RETRO_SDL_VERSION STREQUAL "2")
