diff --git a/package/lirc-tools/lirc-tools.mk b/package/lirc-tools/lirc-tools.mk
index a3ad6bd5a09..c30c737db83 100644
--- a/package/lirc-tools/lirc-tools.mk
+++ b/package/lirc-tools/lirc-tools.mk
@@ -18,6 +18,15 @@ LIRC_TOOLS_AUTORECONF = YES
 LIRC_TOOLS_CONF_ENV = XSLTPROC=yes HAVE_WORKING_POLL=yes
 LIRC_TOOLS_CONF_OPTS = --without-x --enable-devinput --enable-uinput
 
+# batocera
+define LIRC_TOOLS_BUILD_CMDS
+	$(TARGET_CONFIGURE_OPTS) CXX="$(TARGET_CXX)" CC="$(TARGET_CC)" \
+		CC_FOR_BUILD="$(TARGET_CC)" GCC_FOR_BUILD="$(TARGET_CC)" \
+		CXX_FOR_BUILD="$(TARGET_CXX)" LD_FOR_BUILD="$(TARGET_LD)" \
+                CROSS_COMPILE="$(STAGING_DIR)/usr/bin/" \
+		$(MAKE) -C $(@D)
+endef
+
 ifeq ($(BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_4),y)
 LIRC_TOOLS_CONF_ENV += \
 	DEVINPUT_HEADER=$(STAGING_DIR)/usr/include/linux/input-event-codes.h
