--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,10 @@
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
 
+# Disable search for boost-cmake
+# Fixes building with Boost >= 1.70.0
+set(Boost_NO_BOOST_CMAKE ON)
+
 option(USE_QT_GUI "Build with Qt GUI login support" OFF)
 if(USE_QT_GUI)
   add_definitions(-DUSE_QT_GUI_LOGIN=1)
@@ -11,7 +15,6 @@
 endif(USE_QT_GUI)
 
 find_package(Boost
-  CONFIG
   REQUIRED
   system
   filesystem
