From 3fd58c7a39a89a74e00c702ac777da2c585919eb Mon Sep 17 00:00:00 2001
From: Juliano Dorigão <jdorigao@gmail.com>
Date: Tue, 18 Jun 2024 09:11:08 -0400
Subject: [PATCH] no busybox link

---
 applets/install.sh | 54 ----------------------------------------------
 1 file changed, 54 deletions(-)

diff --git a/applets/install.sh b/applets/install.sh
index 4158968..aa985c3 100755
--- a/applets/install.sh
+++ b/applets/install.sh
@@ -80,58 +80,4 @@ rm -f "$prefix/bin/busybox" || exit 1
 mkdir -p "$prefix/bin" || exit 1
 install -m 755 busybox "$prefix/bin/busybox" || exit 1

-for i in $h; do
-	appdir=`dirname "$i"`
-	app=`basename "$i"`
-	if [ x"$noclobber" = x"1" ] && ([ -e "$prefix/$i" ] || [ -h "$prefix/$i" ]); then
-		echo "  $prefix/$i already exists"
-		continue
-	fi
-	mkdir -p "$prefix/$appdir" || exit 1
-	if [ x"$scriptwrapper" = x"y" ]; then
-		if [ x"$swrapall" != x"y" ] && [ x"$i" = x"/bin/sh" ]; then
-			ln $linkopts busybox "$prefix/$i" || exit 1
-		else
-			rm -f "$prefix/$i"
-			echo "#!/bin/busybox" >"$prefix/$i"
-			chmod +x "$prefix/$i"
-		fi
-		echo "	$prefix/$i"
-	elif [ x"$binaries" = x"y" ]; then
-		# Copy the binary over rather
-		if [ -e "$sharedlib_dir/$app" ]; then
-			echo "   Copying $sharedlib_dir/$app to $prefix/$i"
-			cp -pPR "$sharedlib_dir/$app" "$prefix/$i" || exit 1
-		else
-			echo "Error: Could not find $sharedlib_dir/$app"
-			exit 1
-		fi
-	else
-		if [ x"$linkopts" = x"-f" ]; then
-			bb_path="$prefix/bin/busybox"
-		else
-			case "$appdir" in
-			/)
-				bb_path="bin/busybox"
-			;;
-			/bin)
-				bb_path="busybox"
-			;;
-			/sbin)
-				bb_path="../bin/busybox"
-			;;
-			/usr/bin | /usr/sbin)
-				bb_path="../../bin/busybox"
-			;;
-			*)
-				echo "Unknown installation directory: $appdir"
-				exit 1
-			;;
-			esac
-		fi
-		echo "  $prefix/$i -> $bb_path"
-		ln $linkopts "$bb_path" "$prefix/$i" || exit 1
-	fi
-done
-
 exit 0
--
2.45.2

