diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 5c09aab..4d83e9a 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -395,9 +395,6 @@ target_include_directories(libatrac9 PUBLIC LibAtrac9/C/src)
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
 option(XXHASH_BUILD_XXHSUM "Build the xxhsum binary" OFF)
 option(XXH_X86DISPATCH_ALLOW_AVX "Allow building XXH3 with AVX even if it crashes on SSE2-Only CPUs" OFF)
-if((NOT APPLE) AND (ARCHITECTURE STREQUAL "x86_64"))
-	set(DISPATCH 1)
-endif()
 add_subdirectory(xxHash/cmake_unofficial EXCLUDE_FROM_ALL)
 if(XXH_X86DISPATCH_ALLOW_AVX)
 	target_compile_definitions(xxhash PRIVATE XXH_X86DISPATCH_ALLOW_AVX)
diff --git a/external/xxHash/Makefile b/external/xxHash/Makefile
index 5b7b8ff..64d41a6 100644
--- a/external/xxHash/Makefile
+++ b/external/xxHash/Makefile
@@ -109,10 +109,6 @@ default: lib xxhsum_and_links
 all: lib xxhsum xxhsum_inlinedXXH
 
 ## xxhsum is the command line interface (CLI)
-ifeq ($(DISPATCH),1)
-xxhsum: CPPFLAGS += -DXXHSUM_DISPATCH=1
-xxhsum: xxh_x86dispatch.o
-endif
 xxhsum: xxhash.o $(XXHSUM_SPLIT_OBJS)
 	$(CC) $(FLAGS) $^ $(LDFLAGS) -o $@$(EXT)
