diff --git a/package/libarchive/libarchive.mk b/package/libarchive/libarchive.mk
index 8568847493..3126dd1558 100644
--- a/package/libarchive/libarchive.mk
+++ b/package/libarchive/libarchive.mk
@@ -13,31 +13,34 @@ LIBARCHIVE_LICENSE_FILES = COPYING
 LIBARCHIVE_CPE_ID_VENDOR = libarchive
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDTAR),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdtar=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdtar=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdtar=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdtar
 endif
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDCPIO),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdcpio
 endif
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDCAT),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdcat=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdcat=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdcat=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdcat
 endif
@@ -144,6 +147,11 @@ else
 LIBARCHIVE_CONF_OPTS += --without-zstd
 endif
 
+# reglinux - fix building with musl
+ifeq ($(BR2_TOOLCHAIN_USES_MUSL),y)
+LIBARCHIVE_CONF_ENV += LIBS="-latomic"
+endif
+
 # The only user of host-libarchive needs zlib support
 HOST_LIBARCHIVE_DEPENDENCIES = host-zlib
 # needed for autoreconf
