diff --git a/package/libarchive/libarchive.mk b/package/libarchive/libarchive.mk
index 8568847493..4ee77da41f 100644
--- a/package/libarchive/libarchive.mk
+++ b/package/libarchive/libarchive.mk
@@ -13,31 +13,34 @@ LIBARCHIVE_LICENSE_FILES = COPYING
 LIBARCHIVE_CPE_ID_VENDOR = libarchive
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDTAR),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdtar=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdtar=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdtar=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdtar
 endif
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDCPIO),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdcpio=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdcpio
 endif
 
 ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDCAT),y)
-ifeq ($(BR2_STATIC_LIBS),y)
-LIBARCHIVE_CONF_OPTS += --enable-bsdcat=static
-else
+# reglinux
+#ifeq ($(BR2_STATIC_LIBS),y)
+#LIBARCHIVE_CONF_OPTS += --enable-bsdcat=static
+#else
 LIBARCHIVE_CONF_OPTS += --enable-bsdcat=shared
-endif
+#endif
 else
 LIBARCHIVE_CONF_OPTS += --disable-bsdcat
 endif
@@ -71,17 +74,19 @@ else
 LIBARCHIVE_CONF_OPTS += --without-bz2lib
 endif
 
-ifeq ($(BR2_PACKAGE_EXPAT),y)
-LIBARCHIVE_DEPENDENCIES += expat
-else
+# reglinux no need for expat
+#ifeq ($(BR2_PACKAGE_EXPAT),y)
+#LIBARCHIVE_DEPENDENCIES += expat
+#else
 LIBARCHIVE_CONF_OPTS += --without-expat
-endif
+#endif
 
-ifeq ($(BR2_PACKAGE_LIBICONV),y)
-LIBARCHIVE_DEPENDENCIES += libiconv
-else
+# reglinux no need for libiconv
+#ifeq ($(BR2_PACKAGE_LIBICONV),y)
+#LIBARCHIVE_DEPENDENCIES += libiconv
+#else
 LIBARCHIVE_CONF_OPTS += --without-libiconv-prefix
-endif
+#endif
 
 ifeq ($(BR2_PACKAGE_LIBXML2),y)
 LIBARCHIVE_DEPENDENCIES += libxml2
@@ -103,25 +108,28 @@ else
 LIBARCHIVE_CONF_OPTS += --without-lzo2
 endif
 
-ifeq ($(BR2_PACKAGE_MBEDTLS),y)
-LIBARCHIVE_DEPENDENCIES += mbedtls
-LIBARCHIVE_CONF_OPTS += --with-mbedtls
-else
+# reglinux no need for crypto
+#ifeq ($(BR2_PACKAGE_MBEDTLS),y)
+#LIBARCHIVE_DEPENDENCIES += mbedtls
+#LIBARCHIVE_CONF_OPTS += --with-mbedtls
+#else
 LIBARCHIVE_CONF_OPTS += --without-mbedtls
-endif
+#endif
 
-ifeq ($(BR2_PACKAGE_NETTLE),y)
-LIBARCHIVE_DEPENDENCIES += nettle
-LIBARCHIVE_CONF_OPTS += --with-nettle
-else
+# reglinux no need for crypto
+#ifeq ($(BR2_PACKAGE_NETTLE),y)
+#LIBARCHIVE_DEPENDENCIES += nettle
+#LIBARCHIVE_CONF_OPTS += --with-nettle
+#else
 LIBARCHIVE_CONF_OPTS += --without-nettle
-endif
+#endif
 
-ifeq ($(BR2_PACKAGE_OPENSSL),y)
-LIBARCHIVE_DEPENDENCIES += openssl
-else
+# reglinux no need for crypto
+#ifeq ($(BR2_PACKAGE_OPENSSL),y)
+#LIBARCHIVE_DEPENDENCIES += openssl
+#else
 LIBARCHIVE_CONF_OPTS += --without-openssl
-endif
+#endif
 
 ifeq ($(BR2_PACKAGE_ZLIB),y)
 LIBARCHIVE_DEPENDENCIES += zlib
@@ -144,6 +152,11 @@ else
 LIBARCHIVE_CONF_OPTS += --without-zstd
 endif
 
+# reglinux - fix building with musl
+ifeq ($(BR2_TOOLCHAIN_USES_MUSL),y)
+LIBARCHIVE_CONF_ENV += LIBS="-latomic"
+endif
+
 # The only user of host-libarchive needs zlib support
 HOST_LIBARCHIVE_DEPENDENCIES = host-zlib
 # needed for autoreconf
