diff --git a/package/llvm-project/clang/clang.hash b/package/llvm-project/clang/clang.hash
index 33b50af6bf..6f84a1bcc5 100644
--- a/package/llvm-project/clang/clang.hash
+++ b/package/llvm-project/clang/clang.hash
@@ -1,3 +1,4 @@
 # locally calculated
+sha256  11e5e4ecab5338b9914de3b83a4622cb200de466b7c56ba675afb72fa7d64675  clang-19.1.7.src.tar.xz
 sha256  a6b673ef15377fb46062d164e8ddc4d05c348ff8968f015f7f4af03f51000067  clang-15.0.7.src.tar.xz
 sha256  ebcd9bbf783a73d05c53ba4d586b8d5813dcdf3bbec50265860ccc885e606f47  LICENSE.TXT
