diff --git a/dlls/appwiz.cpl/appwiz.c b/dlls/appwiz.cpl/appwiz.c
index b295e40..0272c9a 100644
--- a/dlls/appwiz.cpl/appwiz.c
+++ b/dlls/appwiz.cpl/appwiz.c
@@ -952,12 +952,12 @@
         return FALSE;
 
     if(!wcscmp(params, L"install_gecko")) {
-        install_addon(ADDON_GECKO);
+        /*install_addon(ADDON_GECKO);*/
         return TRUE;
     }
 
     if(!wcscmp(params, L"install_mono")) {
-        install_addon(ADDON_MONO);
+        /*install_addon(ADDON_MONO);*/
         return TRUE;
     }
 
