diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index 6be9289c..d7a1b3a7 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -824,9 +824,9 @@
 
 		Uint32 window_flags = SDL_GetWindowFlags(window);
 		if (g_Config.bFullScreen) {
-			window_flags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+			window_flags |= SDL_WINDOW_FULLSCREEN;
 		} else {
-			window_flags &= ~SDL_WINDOW_FULLSCREEN_DESKTOP;
+			window_flags &= ~SDL_WINDOW_FULLSCREEN;
 		}
 		SDL_SetWindowFullscreen(window, window_flags);
 	}
@@ -956,7 +956,7 @@
 			Native_NotifyWindowHidden(false);
 
 			Uint32 window_flags = SDL_GetWindowFlags(window);
-			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN_DESKTOP));
+			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN));
 
 			// This one calls NativeResized if the size changed.
 			Native_UpdateScreenScale(new_width_px, new_height_px, UIScaleFactorToMultiplier(g_Config.iUIScaleFactor));
@@ -986,7 +986,7 @@
 		case SDL_WINDOWEVENT_MOVED:
 		{
 			Uint32 window_flags = SDL_GetWindowFlags(window);
-			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN_DESKTOP));
+			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN));
 			if (!fullscreen) {
 				g_Config.iWindowX = (int)event.window.data1;
 				g_Config.iWindowY = (int)event.window.data2;
@@ -1466,7 +1466,7 @@
 	Uint32 mode = 0;
 	for (int i = 1; i < argc; i++) {
 		if (!strcmp(argv[i], "--fullscreen")) {
-			mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+			mode |= SDL_WINDOW_FULLSCREEN;
 			g_Config.DoNotSaveSetting(&g_Config.bFullScreen);
 		} else if (set_xres == -2)
 			set_xres = parseInt(argv[i]);
@@ -1634,7 +1634,7 @@
 
 	// Use the setting from the config when initing the window.
 	if (g_Config.bFullScreen) {
-		mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+		mode |= SDL_WINDOW_FULLSCREEN;
 		g_display.pixel_xres = g_DesktopWidth;
 		g_display.pixel_yres = g_DesktopHeight;
 	}
