diff --git a/Core/SaveState.cpp b/Core/SaveState.cpp
index 08f5052b..b7a1fb08 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -338,7 +338,7 @@
 
 	// The prefix is always based on GenerateFullDiscId. So we can find these by scanning, too.
 	std::string GenerateSaveSlotFilename(std::string_view gamePrefix, int slot, const char *extension) {
-		return StringFromFormat("%.*s_%d.%s", STR_VIEW(gamePrefix), slot, extension);
+		return StringFromFormat("%s_%d.%s", STR_VIEW(gamePrefix), slot, extension);
 	}
 
 	Path GenerateSaveSlotPath(std::string_view gamePrefix, int slot, const char *extension) {
--- a/Core/Util/PathUtil.cpp
+++ b/Core/Util/PathUtil.cpp
@@ -657,7 +657,7 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return memStickDirectory / "PPSSPP_STATE";
+		return memStickDirectory;
 	case DIRECTORY_CACHE:
 		return Path("/userdata/system/cache/ppsspp/");
 	case DIRECTORY_TEXTURES:
--- SaveState.cpp.orig	2026-03-03 19:31:43.283241758 +0100
+++ SaveState.cpp	2026-03-03 19:33:28.883175873 +0100
