diff --git a/linux/linux.hash b/linux/linux.hash
index 4bb0e27c90..fc12ae3aa4 100644
--- a/linux/linux.hash
+++ b/linux/linux.hash
@@ -1,7 +1,9 @@
 # From https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
-sha256  e8a099182562aecff781de72ce769461e706d97af42d740dff20eb450dd5771e  linux-6.12.28.tar.xz
-sha256  ff856748671629c1fefef219099e0b4b81131c2d325e768cb0806e204157014e  linux-6.6.90.tar.xz
-sha256  e319a5bb9049ba9fb8cbc08cba4874716e8985bd10f7971f2573ea802c257911  linux-6.1.138.tar.xz
+sha256  8112202bc26d086957a94d2109a6dcd4478c5ba18d0f0a5e1c5dfeea01f54972  linux-6.14.7.tar.xz
+sha256  611fbb669cf5539da48137a9c052becd499f3f862afaf21ab84cea3966221242  linux-6.13.12.tar.xz
+sha256  e8b2ec7e2338ccb9c86de7154f6edcaadfce80907493c143e85a82776bb5064d  linux-6.12.29.tar.xz
+sha256  d08d3d175407a52cd0b25fc95e149bbd2fd6922cd37816c8fcfad18f95e254f4  linux-6.6.91.tar.xz
+sha256  5cb22d2e77af1ae942bf20aa9fcc4901a51cdd25522a30a3beee9140a1c84bc2  linux-6.1.139.tar.gz
 # From https://www.kernel.org/pub/linux/kernel/v5.x/sha256sums.asc
 sha256  b6abfa53315a04e459070b927c58beb41f085433117d58756504d68b67f6a31e  linux-5.15.182.tar.xz
 sha256  1fc82737e0e4f0ae0b40f91a00c68f3cb75362e6ae54623ec64b41a51cde0f25  linux-5.10.237.tar.xz
