diff --git a/linux/linux.hash b/linux/linux.hash
index 56a7b436e0..da62aada4a 100644
--- a/linux/linux.hash
+++ b/linux/linux.hash
@@ -1,10 +1,9 @@
 # From https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
-# From https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
-# From https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
-# From https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
-sha256  4059d394cbf8e9548df36d37e0b8a80c4409ac4e14ecc5019a72a770ef7b41ba  linux-6.12.73.tar.xz
-sha256  16742f4e78abfec8cc9205fb3cf79d63c7e819c33bfc691402683402c350ee2f  linux-6.6.126.tar.xz
-sha256  fd2d033321bd15e0ad5669208b6e43f3f93ccecb059a512ca6b913ca940c38ea  linux-6.1.163.tar.xz
+sha256  02c1878b83f5f9eb0f08dc3852933ed6db2ff31a5f3e908674e1795cdf60952e  linux-6.19.7.tar.xz
+sha256  2bff97ab0590bc5ccdd5b82286d5ef324828672126ebddc88d062e3ac65af80d  linux-6.18.17.tar.xz
+sha256  bbb43e834c46e6bd49a5c28f22e679a937443404e1f653204d4b24929f3ad896  linux-6.12.76.tar.xz
+sha256  caa08f0122224fbbfab177e2a37cc2a94a0046bd2e7e87f03f8913f2b812448a  linux-6.6.129.tar.xz
+sha256  818dfc742158d18f70a6abccd77b7e4366cb7f26a88687f94dae392a9f089449  linux-6.1.166.tar.xz
 # From https://www.kernel.org/pub/linux/kernel/v5.x/sha256sums.asc
 sha256  fce4b1c86688880932ba8f755880cbf390a89453464bfd90b9a1b01a121c2998  linux-5.15.200.tar.xz
 sha256  47754da223a9f264b917be5d575a4dae03fc8777aa9e1b00473e973ee997d529  linux-5.10.250.tar.xz
