--- a/drivers/gpu/drm/drm_edid.c	2024-04-13 12:07:41.000000000 +0100
+++ a/drivers/gpu/drm/drm_edid.c	2024-06-26 00:21:30.180353959 +0100
@@ -3025,7 +3025,7 @@
 {
 	BUILD_BUG_ON(offsetof(typeof(*descriptor), pixel_clock) != 0);
 
-	return descriptor->pixel_clock != 0;
+	return (descriptor->pixel_clock > 0 && descriptor->pixel_clock < 15000);
 }
 
 typedef void detailed_cb(const struct detailed_timing *timing, void *closure);
