--- a/gfx/common/wayland_common.c
+++ b/gfx/common/wayland_common.c
@@ -923,14 +923,14 @@
     * which is useful for HiDPI and auto selecting a display for fullscreen. */
    if (video_monitor_index == 0 && wl_list_length (&wl->all_outputs) > 1)
    {
-      if (!wl_draw_splash_screen(wl))
-         RARCH_ERR("[Wayland] Failed to draw splash screen.\n");
+      /*if (!wl_draw_splash_screen(wl))
+         RARCH_ERR("[Wayland] Failed to draw splash screen.\n");*/
 
       /* Make sure splash screen is on screen and sized */
-      wl->configured = true;
+      /*wl->configured = true;
 
       while (wl->configured)
-      wl_display_dispatch(wl->input.dpy);
+      wl_display_dispatch(wl->input.dpy);*/
    }
 #endif
 
