diff --git a/docs/retroarch.6 b/docs/retroarch.6
index d38a8ee..9fb5330 100644
--- a/docs/retroarch.6
+++ b/docs/retroarch.6
@@ -22,7 +22,7 @@ It also features a simple built-in UI.
 
 .TP
 \fBLoad content, using a specific libretro core and config file.\fR
-retroarch --config ~/.config/retroarch/retroarch.cfg --libretro /path/to/libretro/core.so /path/to/rom.rom --verbose
+retroarch --config ~/configs/retroarch/retroarch.cfg --libretro /path/to/libretro/core.so /path/to/rom.rom --verbose
 
 .TP
 \fBNo command line options will start RetroArch in menu mode.\fR
diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index e4ffe0f..f7b257d 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -1796,7 +1796,7 @@ static void frontend_unix_get_env(int *argc,
    else if (home)
    {
       size_t _len = strlcpy(base_path, home, sizeof(base_path));
-      strlcpy(base_path + _len, "/.config/retroarch", sizeof(base_path) - _len);
+      strlcpy(base_path + _len, "/configs/retroarch", sizeof(base_path) - _len);
    }
    else
       strlcpy(base_path, "retroarch", sizeof(base_path));
@@ -2390,7 +2390,7 @@ static int frontend_unix_parse_drive_list(void *data, bool load_content)
    else if (home)
    {
       size_t _len = strlcpy(base_path, home, sizeof(base_path));
-      strlcpy(base_path + _len, "/.config/retroarch", sizeof(base_path) - _len);
+      strlcpy(base_path + _len, "/configs/retroarch", sizeof(base_path) - _len);
    }
 #endif
 
diff --git a/retroarch.c b/retroarch.c
index e1e007f..b69e1ab 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -6523,7 +6523,7 @@ static void retroarch_print_help(const char *arg0)
          "                                 "
          "  $XDG_CONFIG_HOME/retroarch/retroarch.cfg,\n"
          "                                 "
-         "  $HOME/.config/retroarch/retroarch.cfg, and\n"
+         "  $HOME/configs/retroarch/retroarch.cfg, and\n"
          "                                 "
          "  $HOME/.retroarch.cfg.\n"
          "                                 "
