--- a/Sources/Jazz2/ContentResolver.cpp	2025-01-10 17:31:01.496941494 +0100
+++ b/Sources/Jazz2/ContentResolver.cpp	2025-01-10 17:39:24.264974219 +0100
@@ -203,56 +203,9 @@
 		_sourcePath = fs::CombinePath(appData, "Source/"_s);
 		_cachePath = fs::CombinePath(appData, "Cache/"_s);
 #elif defined(DEATH_TARGET_UNIX)
-#	if defined(NCINE_PACKAGED_CONTENT_PATH)
-		_contentPath = "Content/"_s;
-#	elif defined(NCINE_OVERRIDE_CONTENT_PATH)
-		_contentPath = NCINE_OVERRIDE_CONTENT_PATH;
-#	else
-		_contentPath = NCINE_INSTALL_PREFIX "/share/" NCINE_LINUX_PACKAGE "/Content/";
-#	endif
-#	if defined(NCINE_PACKAGED_CONTENT_PATH)
-		// If Content is packaged with binaries, always use standard XDG paths for everything else
-		auto localStorage = fs::GetLocalStorage();
-		if (!localStorage.empty()) {
-			auto appData = fs::CombinePath(localStorage, NCINE_LINUX_PACKAGE);
-			_sourcePath = fs::CombinePath(appData, "Source/"_s);
-			_cachePath = fs::CombinePath(appData, "Cache/"_s);
-		} else {
-			_sourcePath = "Source/"_s;
-			_cachePath = "Cache/"_s;
-		}
-#	else
-		if (fs::DirectoryExists(_contentPath)) {
-			// Shared Content exists, try to use standard XDG paths
-			auto localStorage = fs::GetLocalStorage();
-			if (!localStorage.empty()) {
-				// Use "$XDG_DATA_HOME/Jazz² Resurrection/" if exists (for backward compatibility), otherwise "$XDG_DATA_HOME/{NCINE_LINUX_PACKAGE}/"
-				_cachePath = fs::CombinePath(localStorage, "Jazz² Resurrection/Cache/"_s);
-				if (!fs::DirectoryExists(_cachePath)) {
-					auto appData = fs::CombinePath(localStorage, NCINE_LINUX_PACKAGE);
-					_cachePath = fs::CombinePath(appData, "Cache/"_s);
-				}
-			} else {
-				_cachePath = "Cache/"_s;
-			}
-
-			// Prefer system-wide Source only if it exists and the local one doesn't exist
-			_sourcePath = fs::CombinePath(fs::GetDirectoryName(_cachePath), "Source/"_s);
-			if (!fs::FindPathCaseInsensitive(fs::CombinePath(_sourcePath, "Anims.j2a"_s)) &&
-				!fs::FindPathCaseInsensitive(fs::CombinePath(_sourcePath, "AnimsSw.j2a"_s))) {
-				auto systemWideSource = NCINE_INSTALL_PREFIX "/share/" NCINE_LINUX_PACKAGE "/Source/";
-				if (fs::FindPathCaseInsensitive(fs::CombinePath(systemWideSource, "Anims.j2a"_s)) ||
-					fs::FindPathCaseInsensitive(fs::CombinePath(systemWideSource, "AnimsSw.j2a"_s))) {
-					_sourcePath = systemWideSource;
-				}
-			}
-		} else {
-			// Fallback to relative paths
-			_contentPath = "Content/"_s;
-			_sourcePath = "Source/"_s;
-			_cachePath = "Cache/"_s;
-		}
-#	endif
+		_sourcePath = "/userdata/roms/jazz2/";
+		_contentPath = "/usr/share/jazz2/Content/";
+		_cachePath = "/userdata/system/cache/jazz2/";
 #elif defined(DEATH_TARGET_WINDOWS_RT)
 		bool found = false;
 		if (Environment::CurrentDeviceType == DeviceType::Xbox) {
