diff --git a/package/fmt/fmt.mk b/package/fmt/fmt.mk
index 113652e9116..274324810d9 100644
--- a/package/fmt/fmt.mk
+++ b/package/fmt/fmt.mk
@@ -3,8 +3,8 @@
 # fmt
 #
 ################################################################################
-
-FMT_VERSION = 11.0.2
+# REGLINUX bump
+FMT_VERSION = 11.1.3
 FMT_SITE = https://github.com/fmtlib/fmt/releases/download/$(FMT_VERSION)
 FMT_SOURCE = fmt-$(FMT_VERSION).zip
 FMT_LICENSE = MIT with exception
