--- a/src/core/spu.cpp	2024-07-13 19:09:09.741895559 +0200
+++ b/src/core/spu.cpp	2024-07-13 19:12:44.646923924 +0200
@@ -73,7 +73,7 @@
   DMARead = 3
 };
 
-union SPUCNT
+union SpuCnt
 {
   u16 bits;
 
@@ -91,7 +91,7 @@
   BitField<u16, u8, 0, 6> mode;
 };
 
-union SPUSTAT
+union SpuStat
 {
   u16 bits;
 
@@ -362,8 +362,8 @@
   TickCount cpu_ticks_per_spu_tick = 0;
   TickCount cpu_tick_divider = 0;
 
-  SPUCNT SPUCNT = {};
-  SPUSTAT SPUSTAT = {};
+  SpuCnt SPUCNT = {};
+  SpuStat SPUSTAT = {};
 
   TransferControl transfer_control = {};
   u16 transfer_address_reg = 0;
@@ -911,7 +911,7 @@
       DEBUG_LOG("SPU control register <- 0x{:04X}", value);
       GeneratePendingSamples();
 
-      const SPUCNT new_value{value};
+      const SpuCnt new_value{value};
       if (new_value.ram_transfer_mode != s_state.SPUCNT.ram_transfer_mode &&
           new_value.ram_transfer_mode == RAMTransferMode::Stopped)
       {
