diff --git a/package/fontconfig/fontconfig.mk b/package/fontconfig/fontconfig.mk
index 11758a4c80..be003356cb 100644
--- a/package/fontconfig/fontconfig.mk
+++ b/package/fontconfig/fontconfig.mk
@@ -3,8 +3,8 @@
 # fontconfig
 #
 ################################################################################
-
-FONTCONFIG_VERSION = 2.14.2
+# reglinux - update
+FONTCONFIG_VERSION = 2.15.0
 FONTCONFIG_SITE = https://www.freedesktop.org/software/fontconfig/release
 FONTCONFIG_SOURCE = fontconfig-$(FONTCONFIG_VERSION).tar.xz
 # 0001-add-pthread-as-a-dependency-of-a-static-lib.patch
