diff --git a/package/fontconfig/fontconfig.hash b/package/fontconfig/fontconfig.hash
index a4f87caf1e..2b267a306c 100644
--- a/package/fontconfig/fontconfig.hash
+++ b/package/fontconfig/fontconfig.hash
@@ -1,5 +1,5 @@
 # From https://lists.freedesktop.org/archives/fontconfig/2023-January/006950.html
-sha256  dba695b57bce15023d2ceedef82062c2b925e51f5d4cc4aef736cf13f60a468b  fontconfig-2.14.2.tar.xz
+sha256  63a0658d0e06e0fa886106452b58ef04f21f58202ea02a94c39de0d3335d7c0e  fontconfig-2.15.0.tar.xz
 
 # Locally calculated
 sha256  51a51aa9823704fd90bccc616cdd17ebabb5b2b3e9cbde886ca02c7002288067  COPYING
