diff --git a/package/gvfs/gvfs.mk b/package/gvfs/gvfs.mk
index 50efed1abbf..a53bf4f54fe 100644
--- a/package/gvfs/gvfs.mk
+++ b/package/gvfs/gvfs.mk
@@ -71,12 +71,14 @@ else
 GVFS_CONF_OPTS += -Dbluray=false
 endif
 
-ifeq ($(BR2_PACKAGE_LIBCAP)$(BR2_PACKAGE_POLKIT),yy)
-GVFS_DEPENDENCIES += libcap polkit
-GVFS_CONF_OPTS += -Dadmin=true
-else
+# batocera
 GVFS_CONF_OPTS += -Dadmin=false
-endif
+#ifeq ($(BR2_PACKAGE_LIBCAP)$(BR2_PACKAGE_POLKIT),yy)
+#GVFS_DEPENDENCIES += libcap polkit
+#GVFS_CONF_OPTS += -Dadmin=true
+#else
+#GVFS_CONF_OPTS += -Dadmin=false
+#endif
 
 ifeq ($(BR2_PACKAGE_LIBCDIO_PARANOIA)$(BR2_PACKAGE_LIBGUDEV),yy)
 GVFS_DEPENDENCIES += libcdio-paranoia libgudev
