diff --git a/package/libxmlb/libxmlb.mk b/package/libxmlb/libxmlb.mk
index 1d5d339e4f4..ee39e92052b 100644
--- a/package/libxmlb/libxmlb.mk
+++ b/package/libxmlb/libxmlb.mk
@@ -34,3 +34,8 @@ LIBXMLB_CONF_OPTS += -Dlzma=disabled
 endif
 
 $(eval $(meson-package))
+
+# reglinux - appstream dependency
+HOST_LIBXMLB_CONF_OPTS = -Dgtkdoc=false -Dtests=false -Dstemmer=false -Dcli=false -Dlzma=enabled -Dzstd=enabled
+HOST_LIBXMLB_DEPENDENCIES = host-libglib2 host-xz host-gobject-introspection
+$(eval $(host-meson-package))
