--- a/src/osdep/sigsegv_handler.cpp	2025-08-25 22:20:36.070541497 +0200
+++ b/src/osdep/sigsegv_handler.cpp	2025-08-25 22:21:05.414363576 +0200
@@ -36,7 +36,7 @@
 #endif
 #include "uae.h"
 
-#if !defined(__MACH__) && !defined(CPU_AMD64) && !defined(__x86_64__) && !defined(__riscv)
+#if !defined(__MACH__) && !defined(CPU_AMD64) && !defined(__x86_64__) && !defined(__riscv) && defined(__GLIBC__)
 #include <asm/sigcontext.h>
 #else
 #include <sys/ucontext.h>
