--- a/loadzip.cpp	2024-10-22 05:46:40.000000000 +0800
+++ b/loadzip.cpp	2025-10-18 22:56:12.639389318 +0800
@@ -57,6 +57,13 @@
 			break;
 		}
 
+		if (len > 4 && name[len - 4] == '.' && name[len - 3] == 's' && name[len - 2] == 'f' && name[len - 1] == 'c')
+		{
+			strcpy(filename, name);
+			filesize = info.uncompressed_size;
+			break;
+		}
+
 		if (strncasecmp(name, "program.rom", 11) == 0)
 		{
 			strcpy(filename, name);
@@ -68,8 +75,8 @@
 	}
 
 	int len = strlen(zipname);
-	if (!(port == UNZ_END_OF_LIST_OF_FILE || port == UNZ_OK) || filesize == 0 ||
-		(len > 5 && strcasecmp(zipname + len - 5, ".msu1") == 0 && strcasecmp(filename, "program.rom") != 0))
+	if (!(port == UNZ_END_OF_LIST_OF_FILE || port == UNZ_OK) || filesize == 0 /*||
+		(len > 5 && strcasecmp(zipname + len - 5, ".msu1") == 0 && strcasecmp(filename, "program.rom") != 0)*/)
 	{
 		if (unzClose(file) != UNZ_OK)
 			assert(FALSE);
--- a/libretro/libretro.cpp	2025-10-17 00:15:17.555208008 +0800
+++ b/libretro/libretro.cpp	2025-10-18 22:54:56.843855711 +0800
@@ -819,7 +819,7 @@
 #endif
     info->library_version = VERSION GIT_VERSION;
     info->valid_extensions = "smc|sfc|swc|fig|bs|st|zip|msu1";
-    info->need_fullpath = false;
+    info->need_fullpath = true;
     info->block_extract = false;
 }
 
