--- a/scripts/Makefile.extrawarn	2025-01-25 17:37:39.190023225 +0100
+++ b/scripts/Makefile.extrawarn	2025-01-25 17:38:13.705899883 +0100
@@ -43,6 +43,9 @@
 # These result in bogus false positives
 KBUILD_CFLAGS += $(call cc-disable-warning, dangling-pointer)
 
+# Makes GCC 14 happy
+KBUILD_CFLAGS += $(call cc-disable-warning, int-conversion)
+KBUILD_CFLAGS += -Wno-error=suggest-attribute=format
 # Variable Length Arrays (VLAs) should not be used anywhere in the kernel
 KBUILD_CFLAGS += -Wvla
 
