--- a/drivers/gpu/drm/verisilicon/vs_simple_enc.c	2025-04-02 16:44:16.828650489 +0200
+++ b/drivers/gpu/drm/verisilicon/vs_simple_enc.c	2025-04-02 16:44:59.063620555 +0200
@@ -221,6 +221,7 @@
 			drm_of_find_possible_crtcs(drm_dev, dev->of_node);
 	encoder->possible_crtcs = 2;
 
+#ifdef CONFIG_STARFIVE_DSI
 	ret = drm_of_find_panel_or_bridge(dev->of_node, 1, 0,&tmp_panel, &bridge);
 	if (ret){
 		printk("no panel, %d\n",ret);
@@ -229,6 +230,7 @@
 	}
 	if (tmp_panel)
 		dev_err(dev, "found panel on endpoint\n");
+#endif
 
 #if KERNEL_VERSION(5, 7, 0) <= LINUX_VERSION_CODE
 	ret = drm_bridge_attach(encoder, bridge, NULL, 0);
