diff --git a/package/x11r7/xfont_font-util/xfont_font-util.mk b/package/x11r7/xfont_font-util/xfont_font-util.mk
index 8a97c85d684..bc0627cc67e 100644
--- a/package/x11r7/xfont_font-util/xfont_font-util.mk
+++ b/package/x11r7/xfont_font-util/xfont_font-util.mk
@@ -3,8 +3,8 @@
 # xfont_font-util
 #
 ################################################################################
-
-XFONT_FONT_UTIL_VERSION = 1.4.0
+# reglinux - update
+XFONT_FONT_UTIL_VERSION = 1.4.1
 XFONT_FONT_UTIL_SOURCE = font-util-$(XFONT_FONT_UTIL_VERSION).tar.xz
 XFONT_FONT_UTIL_SITE = https://xorg.freedesktop.org/archive/individual/font
 XFONT_FONT_UTIL_LICENSE = MIT/BSD-2-Clause
