diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index d3ed355..6384193 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -5189,44 +5189,8 @@
 static bool locate_amiberry_conf(const bool portable_mode)
 {
 	config_path = get_config_directory(portable_mode);
-#ifdef __MACH__
-	if constexpr (true)
-#elif defined(__ANDROID__)
-	if constexpr (true)
-#elif defined(_WIN32)
-	if constexpr (true)
-#else
-	if (portable_mode)
-#endif
-	{
-		amiberry_conf_file = config_path + "/amiberry.conf";
-		amiberry_ini_file = config_path + "/amiberry.ini";
-	}
-	else
-	{
-#ifdef __MACH__
-		const std::string amiberry_dir = "Amiberry";
-#elif defined(__ANDROID__)
-        const std::string amiberry_dir = "amiberry";
-        // On Android, use the home directory (external files dir) for config
-        config_path = get_home_directory(false);
-        amiberry_conf_file = config_path + "amiberry.conf";
-        amiberry_ini_file = config_path + "amiberry.ini";
-        return my_existsfile2(amiberry_conf_file.c_str());
-#else
-		const std::string amiberry_dir = "amiberry";
-#endif
-		std::string xdg_config_home = get_xdg_config_home();
-		if (!my_existsdir(xdg_config_home.c_str()))
-			my_mkdir(xdg_config_home.c_str());
-		xdg_config_home += "/" + amiberry_dir;
-		if (!my_existsdir(xdg_config_home.c_str()))
-			my_mkdir(xdg_config_home.c_str());
-
-		amiberry_conf_file = xdg_config_home + "/amiberry.conf";
-		amiberry_ini_file = xdg_config_home + "/amiberry.ini";
-        config_path = xdg_config_home;
-	}
+	amiberry_conf_file = std::string("/userdata/system/configs/amiberry/conf/amiberry.conf");
+	amiberry_ini_file = std::string("/userdata/system/configs/amiberry/conf/amiberry.ini");
 	return my_existsfile2(amiberry_conf_file.c_str());
 }
 
