diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5425119..a26505b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,9 @@ SET(LIB_MINOR_VERSION "24")
 SET(LIB_REVISION "11")
 SET(LIB_VERSION "${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_REVISION}")
 
+# Enable PIE
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
 # Build Type
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
 	# By default, we use Release
