From b3b2b643c9d3bd98ade7fda3e56d47671121f766 Mon Sep 17 00:00:00 2001
From: Antonio Giner <calamity15khz@gmail.com>
Date: Sun, 12 May 2024 21:06:55 +0200
Subject: [PATCH] Fix 2 buffer overflows.

---
 switchres.cpp      | 2 +-
 switchres_main.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/switchres.cpp b/switchres.cpp
index fb68f07..5623117 100644
--- a/switchres.cpp
+++ b/switchres.cpp
@@ -89,7 +89,7 @@ switchres_manager::switchres_manager()
 	display()->set_monitor("generic_15");
 	display()->set_modeline("auto");
 	display()->set_lcd_range("auto");
-	for (int i = 0; i++ < MAX_RANGES;) display()->set_crt_range(i, "auto");
+	for (int i = 0; i < MAX_RANGES; i++) display()->set_crt_range(i, "auto");
 	display()->set_screen("auto");
 	display()->set_modeline_generation(true);
 	display()->set_lock_unsupported_modes(true);
diff --git a/switchres_main.cpp b/switchres_main.cpp
index 7413fbd..ac16a94 100644
--- a/switchres_main.cpp
+++ b/switchres_main.cpp
@@ -375,7 +375,7 @@ int main(int argc, char **argv)
 				monitor_range *range = &switchres.display()->range[mode->range];
 				edid_from_modeline(mode, range, switchres.display()->monitor(), &edid);
 
-				char file_name[strlen(switchres.display()->monitor()) + 4];
+				char file_name[strlen(switchres.display()->monitor()) + 5];
 				sprintf(file_name, "%s.bin", switchres.display()->monitor());
 
 				FILE *file = fopen(file_name, "wb");
