diff --git a/package/x11r7/xcb-proto/xcb-proto.mk b/package/x11r7/xcb-proto/xcb-proto.mk
index 324b29427dd..b5f72721a34 100644
--- a/package/x11r7/xcb-proto/xcb-proto.mk
+++ b/package/x11r7/xcb-proto/xcb-proto.mk
@@ -3,8 +3,8 @@
 # xcb-proto
 #
 ################################################################################
-
-XCB_PROTO_VERSION = 1.15.2
+# reglinux - update
+XCB_PROTO_VERSION = 1.16.0
 XCB_PROTO_SOURCE = xcb-proto-$(XCB_PROTO_VERSION).tar.xz
 XCB_PROTO_SITE = https://xorg.freedesktop.org/archive/individual/proto
 XCB_PROTO_LICENSE = MIT
