--- a/Cargo.toml	2025-11-03 01:19:00.200839204 +0700
+++ b/Cargo.toml	2025-11-03 01:19:17.139769564 +0700
@@ -44,7 +44,7 @@
 
 [profile.release]
 panic = "abort"
-lto = "thin"
+lto = "auto"
 
 [profile.release-with-debug]
 inherits = "release"
--- a/build.rs	2025-11-03 01:20:20.469529399 +0700
+++ b/build.rs	2025-11-03 01:20:42.109515959 +0700
@@ -95,9 +95,9 @@
             .unwrap();
     }
 
-    volk_build.flag("-flto=thin");
-    rdp_build.flag("-flto=thin");
-    simd_build.flag("-flto=thin");
+    volk_build.flag("-flto=auto");
+    rdp_build.flag("-flto=auto");
+    simd_build.flag("-flto=auto");
 
     volk_build.compile("volk");
     rdp_build.compile("parallel-rdp");
