--- a/src/renderercommon/tr_image_svg.c	2024-04-17 12:08:40.827166740 +0200
+++ b/src/renderercommon/tr_image_svg.c	2024-04-17 12:11:33.766888427 +0200
@@ -115,7 +115,7 @@
 	columns = (int)(image->width * scale);
 	rows    = (int)(image->height * scale);
 
-#ifdef __ANDROID__
+#if defined(__ANDROID__) || defined(__aarch64__)
 	if (!Com_PowerOf2(columns) || !Com_PowerOf2(rows))
 #else
 	if (!GLEW_ARB_texture_non_power_of_two && (!Com_PowerOf2(columns) || !Com_PowerOf2(rows)))
