diff --git a/package/spdlog/spdlog.hash b/package/spdlog/spdlog.hash
index 69e569a548..8c6838160b 100644
--- a/package/spdlog/spdlog.hash
+++ b/package/spdlog/spdlog.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  9962648c9b4f1a7bbc76fd8d9172555bad1871fdb14ff4f842ef87949682caa5  spdlog-1.15.0.tar.gz
+sha256  25c843860f039a1600f232c6eb9e01e6627f7d030a2ae5e232bdd3c9205d26cc  spdlog-1.15.1.tar.gz
 sha256  4ccecab18d1ff0b61174fe3d6c430541625d3ddb865b0d5887db296f883c76e7  LICENSE
