--- a/launcher/CMakeLists.txt	2024-05-14 00:22:45.583415339 +0200
+++ b/launcher/CMakeLists.txt	2024-05-14 00:22:59.623750869 +0200
@@ -109,7 +109,7 @@
 
 if(TARGET Qt6::Core)
 	if(ENABLE_TRANSLATIONS)
-		set_source_files_properties(${launcher_TS} PROPERTIES OUTPUT_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/translation)
+		#set_source_files_properties(${launcher_TS} PROPERTIES OUTPUT_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/translation)
 		qt_add_translations(vcmilauncher
 			TS_FILES ${launcher_TS}
 			QM_FILES_OUTPUT_VARIABLE launcher_QM
