diff --git a/Makefile.common b/Makefile.common
index 72aef3a..f3e5a7e 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -467,7 +467,7 @@ endif
 
 SOURCES_C += $(LIBRETRO_COMM_DIR)/libco/libco.c
 
-ifeq ($(platform), $(filter $(platform), rpi3 rpi4 libnx jetson-xavier))
+ifeq ($(platform), $(filter $(platform), rpi3 rpi4 libnx jetson-xavier arm64))
 	SOURCES_CXX += $(VIDEODIR_GLIDEN64)/src/CRC32_ARMV8.cpp
 else
 	SOURCES_CXX += $(VIDEODIR_GLIDEN64)/src/CRC_OPT.cpp
