diff --git a/package/vulkan-loader/vulkan-loader.hash b/package/vulkan-loader/vulkan-loader.hash
index e09ecda8a9d..864d35509a7 100644
--- a/package/vulkan-loader/vulkan-loader.hash
+++ b/package/vulkan-loader/vulkan-loader.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  3bbaa5ee64058a89949eb777de66ce94bfe3141892514172cfc9451c756802d5  vulkan-loader-1.3.262.tar.gz
+sha256  368d8281604a3f4dee038bfcc55c44e305031ec67f6e3fdd50cdeb83586c99f9  vulkan-loader-v1.4.304.tar.gz
 sha256  43c0a37e6a0fa7ff3c843b3ec5a4fac84b712558ddac103fbd4c1649662a9ece  LICENSE.txt
