diff --git a/package/vulkan-loader/vulkan-loader.mk b/package/vulkan-loader/vulkan-loader.mk
index 7aceb207b5..04286ebb68 100644
--- a/package/vulkan-loader/vulkan-loader.mk
+++ b/package/vulkan-loader/vulkan-loader.mk
@@ -4,8 +4,8 @@
 #
 ################################################################################
 
-VULKAN_LOADER_VERSION = $(VULKAN_HEADERS_VERSION)
-VULKAN_LOADER_SITE = $(call github,KhronosGroup,Vulkan-Loader,v$(VULKAN_LOADER_VERSION))
+VULKAN_LOADER_VERSION = v1.4.304
+VULKAN_LOADER_SITE = $(call github,KhronosGroup,Vulkan-Loader,$(VULKAN_LOADER_VERSION))
 VULKAN_LOADER_LICENSE = Apache-2.0
 VULKAN_LOADER_LICENSE_FILES = LICENSE.txt
 VULKAN_LOADER_INSTALL_STAGING = YES
@@ -15,9 +15,15 @@ VULKAN_LOADER_DEPENDENCIES = host-pkgconf vulkan-headers
 VULKAN_LOADER_CONF_OPTS += \
 	-DASSEMBLER_WORKS=FALSE \
 	-DBUILD_WSI_SCREEN_QNX_SUPPORT=OFF \
-	-DBUILD_WSI_DIRECTFB_SUPPORT=OFF \
 	-DLOADER_CODEGEN=OFF
 
+ifeq ($(BR2_PACKAGE_DIRECTFB),y)
+VULKAN_LOADER_DEPENDENCIES += directfb
+VULKAN_LOADER_CONF_OPTS += -DBUILD_WSI_DIRECTFB_SUPPORT=ON
+else
+VULKAN_LOADER_CONF_OPTS += -DBUILD_WSI_DIRECTFB_SUPPORT=OFF
+endif
+
 ifeq ($(BR2_PACKAGE_LIBXCB),y)
 VULKAN_LOADER_DEPENDENCIES += libxcb
 VULKAN_LOADER_CONF_OPTS += -DBUILD_WSI_XCB_SUPPORT=ON
